From d72498fad93abf1817c4998d172a3d7f98f76bbe Mon Sep 17 00:00:00 2001
From: Shuming Fan <shumingf@realtek.com>
Date: Sat, 14 Sep 2024 17:05:21 +0800
Subject: [PATCH] ASoC: rt1320: reads patch code from firmware file
Git-commit: d72498fad93abf1817c4998d172a3d7f98f76bbe
Patch-mainline: v6.13-rc1
References: jsc#PED-14430 jsc#PED-14297 jsc#PED-14024

This patch removes many lines of the patch code and
reads the patch code from firmware files.

Signed-off-by: Shuming Fan <shumingf@realtek.com>
Link: https://patch.msgid.link/20240914090521.2224276-1-shumingf@realtek.com
Signed-off-by: Mark Brown <broonie@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 sound/soc/codecs/rt1320-sdw.c | 3347 ++-------------------------------
 sound/soc/codecs/rt1320-sdw.h |    2 +
 2 files changed, 107 insertions(+), 3242 deletions(-)

diff --git a/sound/soc/codecs/rt1320-sdw.c b/sound/soc/codecs/rt1320-sdw.c
index f4e1ea29c265..2404de8ae03d 100644
--- a/sound/soc/codecs/rt1320-sdw.c
+++ b/sound/soc/codecs/rt1320-sdw.c
@@ -89,6 +89,23 @@ static const struct reg_sequence rt1320_blind_write[] = {
 	{ 0xc019, 0x10 },
 	{ 0xd487, 0x3f },
 	{ 0xd486, 0xc3 },
+	{ 0x3fc2bfc7, 0x00 },
+	{ 0x3fc2bfc6, 0x00 },
+	{ 0x3fc2bfc5, 0x00 },
+	{ 0x3fc2bfc4, 0x01 },
+	{ 0x0000d486, 0x43 },
+	{ 0x1000db00, 0x02 },
+	{ 0x1000db01, 0x00 },
+	{ 0x1000db02, 0x11 },
+	{ 0x1000db03, 0x00 },
+	{ 0x1000db04, 0x00 },
+	{ 0x1000db05, 0x82 },
+	{ 0x1000db06, 0x04 },
+	{ 0x1000db07, 0xf1 },
+	{ 0x1000db08, 0x00 },
+	{ 0x1000db09, 0x00 },
+	{ 0x1000db0a, 0x40 },
+	{ 0x0000d540, 0x01 },
 };
 
 static const struct reg_sequence rt1320_vc_blind_write[] = {
@@ -148,6 +165,12 @@ static const struct reg_sequence rt1320_vc_blind_write[] = {
 	{ 0xd487, 0x3b },
 	{ 0xd486, 0xc3 },
 	{ 0xc598, 0x04 },
+	{ 0xdb03, 0xf0 },
+	{ 0xdb09, 0x00 },
+	{ 0xdb08, 0x7a },
+	{ 0xdb19, 0x02 },
+	{ 0xdb07, 0x5a },
+	{ 0xdb05, 0x45 },
 	{ 0xd500, 0x00 },
 	{ 0xd500, 0x17 },
 	{ 0xd600, 0x01 },
@@ -164,1913 +187,6 @@ static const struct reg_sequence rt1320_vc_blind_write[] = {
 	{ 0xd610, 0x01 },
 	{ 0xd608, 0x03 },
 	{ 0xd609, 0x00 },
-};
-
-static const struct reg_sequence rt1320_vc_patch_code_write[] = {
-	{ 0x10007000, 0x37 },
-	{ 0x10007001, 0x77 },
-	{ 0x10007002, 0x00 },
-	{ 0x10007003, 0x10 },
-	{ 0x10007004, 0xb7 },
-	{ 0x10007005, 0xe7 },
-	{ 0x10007006, 0x00 },
-	{ 0x10007007, 0x10 },
-	{ 0x10007008, 0x13 },
-	{ 0x10007009, 0x07 },
-	{ 0x1000700a, 0x87 },
-	{ 0x1000700b, 0x48 },
-	{ 0x1000700c, 0x23 },
-	{ 0x1000700d, 0xa6 },
-	{ 0x1000700e, 0xe7 },
-	{ 0x1000700f, 0xee },
-	{ 0x10007010, 0x37 },
-	{ 0x10007011, 0x77 },
-	{ 0x10007012, 0x00 },
-	{ 0x10007013, 0x10 },
-	{ 0x10007014, 0x13 },
-	{ 0x10007015, 0x07 },
-	{ 0x10007016, 0x87 },
-	{ 0x10007017, 0x56 },
-	{ 0x10007018, 0x23 },
-	{ 0x10007019, 0xac },
-	{ 0x1000701a, 0xe7 },
-	{ 0x1000701b, 0xde },
-	{ 0x1000701c, 0x37 },
-	{ 0x1000701d, 0x77 },
-	{ 0x1000701e, 0x00 },
-	{ 0x1000701f, 0x10 },
-	{ 0x10007020, 0x13 },
-	{ 0x10007021, 0x07 },
-	{ 0x10007022, 0xc7 },
-	{ 0x10007023, 0x5f },
-	{ 0x10007024, 0x23 },
-	{ 0x10007025, 0xae },
-	{ 0x10007026, 0xe7 },
-	{ 0x10007027, 0xdc },
-	{ 0x10007028, 0x37 },
-	{ 0x10007029, 0x87 },
-	{ 0x1000702a, 0x00 },
-	{ 0x1000702b, 0x10 },
-	{ 0x1000702c, 0x13 },
-	{ 0x1000702d, 0x07 },
-	{ 0x1000702e, 0xc7 },
-	{ 0x1000702f, 0x86 },
-	{ 0x10007030, 0x23 },
-	{ 0x10007031, 0xae },
-	{ 0x10007032, 0xe7 },
-	{ 0x10007033, 0xe6 },
-	{ 0x10007034, 0x37 },
-	{ 0x10007035, 0x77 },
-	{ 0x10007036, 0x00 },
-	{ 0x10007037, 0x10 },
-	{ 0x10007038, 0x13 },
-	{ 0x10007039, 0x07 },
-	{ 0x1000703a, 0x07 },
-	{ 0x1000703b, 0x40 },
-	{ 0x1000703c, 0x23 },
-	{ 0x1000703d, 0xa6 },
-	{ 0x1000703e, 0xe7 },
-	{ 0x1000703f, 0xe8 },
-	{ 0x10007040, 0x37 },
-	{ 0x10007041, 0x77 },
-	{ 0x10007042, 0x00 },
-	{ 0x10007043, 0x10 },
-	{ 0x10007044, 0x13 },
-	{ 0x10007045, 0x07 },
-	{ 0x10007046, 0xc7 },
-	{ 0x10007047, 0x63 },
-	{ 0x10007048, 0x23 },
-	{ 0x10007049, 0xa2 },
-	{ 0x1000704a, 0xe7 },
-	{ 0x1000704b, 0xec },
-	{ 0x1000704c, 0x37 },
-	{ 0x1000704d, 0x77 },
-	{ 0x1000704e, 0x00 },
-	{ 0x1000704f, 0x10 },
-	{ 0x10007050, 0x13 },
-	{ 0x10007051, 0x07 },
-	{ 0x10007052, 0x47 },
-	{ 0x10007053, 0x6f },
-	{ 0x10007054, 0x23 },
-	{ 0x10007055, 0xa6 },
-	{ 0x10007056, 0xe7 },
-	{ 0x10007057, 0xec },
-	{ 0x10007058, 0x37 },
-	{ 0x10007059, 0x77 },
-	{ 0x1000705a, 0x00 },
-	{ 0x1000705b, 0x10 },
-	{ 0x1000705c, 0x13 },
-	{ 0x1000705d, 0x07 },
-	{ 0x1000705e, 0x07 },
-	{ 0x1000705f, 0x44 },
-	{ 0x10007060, 0x23 },
-	{ 0x10007061, 0xa8 },
-	{ 0x10007062, 0xe7 },
-	{ 0x10007063, 0xec },
-	{ 0x10007064, 0x37 },
-	{ 0x10007065, 0x87 },
-	{ 0x10007066, 0x00 },
-	{ 0x10007067, 0x10 },
-	{ 0x10007068, 0x13 },
-	{ 0x10007069, 0x07 },
-	{ 0x1000706a, 0x87 },
-	{ 0x1000706b, 0x84 },
-	{ 0x1000706c, 0x23 },
-	{ 0x1000706d, 0xa8 },
-	{ 0x1000706e, 0xe7 },
-	{ 0x1000706f, 0xee },
-	{ 0x10007070, 0x37 },
-	{ 0x10007071, 0x87 },
-	{ 0x10007072, 0x00 },
-	{ 0x10007073, 0x10 },
-	{ 0x10007074, 0x13 },
-	{ 0x10007075, 0x07 },
-	{ 0x10007076, 0x47 },
-	{ 0x10007077, 0x97 },
-	{ 0x10007078, 0x23 },
-	{ 0x10007079, 0xaa },
-	{ 0x1000707a, 0xe7 },
-	{ 0x1000707b, 0xee },
-	{ 0x1000707c, 0x67 },
-	{ 0x1000707d, 0x80 },
-	{ 0x1000707e, 0x00 },
-	{ 0x1000707f, 0x00 },
-	{ 0x10007400, 0xb7 },
-	{ 0x10007401, 0xd6 },
-	{ 0x10007402, 0x00 },
-	{ 0x10007403, 0x00 },
-	{ 0x10007404, 0x83 },
-	{ 0x10007405, 0xc7 },
-	{ 0x10007406, 0x06 },
-	{ 0x10007407, 0x47 },
-	{ 0x10007408, 0x93 },
-	{ 0x10007409, 0xf7 },
-	{ 0x1000740a, 0x87 },
-	{ 0x1000740b, 0x00 },
-	{ 0x1000740c, 0x63 },
-	{ 0x1000740d, 0x88 },
-	{ 0x1000740e, 0x07 },
-	{ 0x1000740f, 0x02 },
-	{ 0x10007410, 0x03 },
-	{ 0x10007411, 0xc7 },
-	{ 0x10007412, 0x31 },
-	{ 0x10007413, 0x43 },
-	{ 0x10007414, 0x63 },
-	{ 0x10007415, 0x14 },
-	{ 0x10007416, 0x07 },
-	{ 0x10007417, 0x02 },
-	{ 0x10007418, 0x13 },
-	{ 0x10007419, 0x07 },
-	{ 0x1000741a, 0x10 },
-	{ 0x1000741b, 0x00 },
-	{ 0x1000741c, 0xa3 },
-	{ 0x1000741d, 0x89 },
-	{ 0x1000741e, 0xe1 },
-	{ 0x1000741f, 0x42 },
-	{ 0x10007420, 0x37 },
-	{ 0x10007421, 0xc7 },
-	{ 0x10007422, 0x00 },
-	{ 0x10007423, 0x00 },
-	{ 0x10007424, 0x03 },
-	{ 0x10007425, 0x46 },
-	{ 0x10007426, 0x07 },
-	{ 0x10007427, 0x06 },
-	{ 0x10007428, 0x23 },
-	{ 0x10007429, 0x8a },
-	{ 0x1000742a, 0xc1 },
-	{ 0x1000742b, 0x42 },
-	{ 0x1000742c, 0x83 },
-	{ 0x1000742d, 0xc7 },
-	{ 0x1000742e, 0x46 },
-	{ 0x1000742f, 0x47 },
-	{ 0x10007430, 0x93 },
-	{ 0x10007431, 0xf7 },
-	{ 0x10007432, 0xf7 },
-	{ 0x10007433, 0x0f },
-	{ 0x10007434, 0x23 },
-	{ 0x10007435, 0x00 },
-	{ 0x10007436, 0xf7 },
-	{ 0x10007437, 0x06 },
-	{ 0x10007438, 0x23 },
-	{ 0x10007439, 0x89 },
-	{ 0x1000743a, 0x01 },
-	{ 0x1000743b, 0x42 },
-	{ 0x1000743c, 0x67 },
-	{ 0x1000743d, 0x80 },
-	{ 0x1000743e, 0x00 },
-	{ 0x1000743f, 0x00 },
-	{ 0x10007440, 0x37 },
-	{ 0x10007441, 0xc7 },
-	{ 0x10007442, 0x00 },
-	{ 0x10007443, 0x00 },
-	{ 0x10007444, 0x83 },
-	{ 0x10007445, 0x27 },
-	{ 0x10007446, 0xc7 },
-	{ 0x10007447, 0x5f },
-	{ 0x10007448, 0x13 },
-	{ 0x10007449, 0x05 },
-	{ 0x1000744a, 0x00 },
-	{ 0x1000744b, 0x00 },
-	{ 0x1000744c, 0x23 },
-	{ 0x1000744d, 0xa2 },
-	{ 0x1000744e, 0xf1 },
-	{ 0x1000744f, 0x42 },
-	{ 0x10007450, 0xb7 },
-	{ 0x10007451, 0x06 },
-	{ 0x10007452, 0x00 },
-	{ 0x10007453, 0x10 },
-	{ 0x10007454, 0xb3 },
-	{ 0x10007455, 0xf7 },
-	{ 0x10007456, 0xd7 },
-	{ 0x10007457, 0x00 },
-	{ 0x10007458, 0x63 },
-	{ 0x10007459, 0x86 },
-	{ 0x1000745a, 0x07 },
-	{ 0x1000745b, 0x02 },
-	{ 0x1000745c, 0x83 },
-	{ 0x1000745d, 0x47 },
-	{ 0x1000745e, 0x07 },
-	{ 0x1000745f, 0x56 },
-	{ 0x10007460, 0x93 },
-	{ 0x10007461, 0xf7 },
-	{ 0x10007462, 0x87 },
-	{ 0x10007463, 0x01 },
-	{ 0x10007464, 0x63 },
-	{ 0x10007465, 0x80 },
-	{ 0x10007466, 0x07 },
-	{ 0x10007467, 0x02 },
-	{ 0x10007468, 0x83 },
-	{ 0x10007469, 0x47 },
-	{ 0x1000746a, 0x17 },
-	{ 0x1000746b, 0x08 },
-	{ 0x1000746c, 0x93 },
-	{ 0x1000746d, 0xf7 },
-	{ 0x1000746e, 0x47 },
-	{ 0x1000746f, 0x00 },
-	{ 0x10007470, 0x63 },
-	{ 0x10007471, 0x8a },
-	{ 0x10007472, 0x07 },
-	{ 0x10007473, 0x00 },
-	{ 0x10007474, 0xb7 },
-	{ 0x10007475, 0xc7 },
-	{ 0x10007476, 0xc2 },
-	{ 0x10007477, 0x3f },
-	{ 0x10007478, 0x03 },
-	{ 0x10007479, 0xa5 },
-	{ 0x1000747a, 0x47 },
-	{ 0x1000747b, 0xfc },
-	{ 0x1000747c, 0x13 },
-	{ 0x1000747d, 0x55 },
-	{ 0x1000747e, 0x25 },
-	{ 0x1000747f, 0x00 },
-	{ 0x10007480, 0x13 },
-	{ 0x10007481, 0x75 },
-	{ 0x10007482, 0x15 },
-	{ 0x10007483, 0x00 },
-	{ 0x10007484, 0x67 },
-	{ 0x10007485, 0x80 },
-	{ 0x10007486, 0x00 },
-	{ 0x10007487, 0x00 },
-	{ 0x10007488, 0x03 },
-	{ 0x10007489, 0xa7 },
-	{ 0x1000748a, 0x81 },
-	{ 0x1000748b, 0x57 },
-	{ 0x1000748c, 0x13 },
-	{ 0x1000748d, 0x01 },
-	{ 0x1000748e, 0x01 },
-	{ 0x1000748f, 0xff },
-	{ 0x10007490, 0x23 },
-	{ 0x10007491, 0x26 },
-	{ 0x10007492, 0x11 },
-	{ 0x10007493, 0x00 },
-	{ 0x10007494, 0x23 },
-	{ 0x10007495, 0x24 },
-	{ 0x10007496, 0x81 },
-	{ 0x10007497, 0x00 },
-	{ 0x10007498, 0x23 },
-	{ 0x10007499, 0x22 },
-	{ 0x1000749a, 0x91 },
-	{ 0x1000749b, 0x00 },
-	{ 0x1000749c, 0x93 },
-	{ 0x1000749d, 0x07 },
-	{ 0x1000749e, 0xa0 },
-	{ 0x1000749f, 0x05 },
-	{ 0x100074a0, 0x63 },
-	{ 0x100074a1, 0x14 },
-	{ 0x100074a2, 0xf7 },
-	{ 0x100074a3, 0x04 },
-	{ 0x100074a4, 0x37 },
-	{ 0x100074a5, 0x07 },
-	{ 0x100074a6, 0x00 },
-	{ 0x100074a7, 0x11 },
-	{ 0x100074a8, 0x83 },
-	{ 0x100074a9, 0x47 },
-	{ 0x100074aa, 0x07 },
-	{ 0x100074ab, 0x01 },
-	{ 0x100074ac, 0x13 },
-	{ 0x100074ad, 0x06 },
-	{ 0x100074ae, 0x30 },
-	{ 0x100074af, 0x00 },
-	{ 0x100074b0, 0x93 },
-	{ 0x100074b1, 0xf7 },
-	{ 0x100074b2, 0xf7 },
-	{ 0x100074b3, 0x0f },
-	{ 0x100074b4, 0x63 },
-	{ 0x100074b5, 0x9a },
-	{ 0x100074b6, 0xc7 },
-	{ 0x100074b7, 0x02 },
-	{ 0x100074b8, 0x03 },
-	{ 0x100074b9, 0x47 },
-	{ 0x100074ba, 0x87 },
-	{ 0x100074bb, 0x01 },
-	{ 0x100074bc, 0x13 },
-	{ 0x100074bd, 0x77 },
-	{ 0x100074be, 0xf7 },
-	{ 0x100074bf, 0x0f },
-	{ 0x100074c0, 0x63 },
-	{ 0x100074c1, 0x14 },
-	{ 0x100074c2, 0xf7 },
-	{ 0x100074c3, 0x02 },
-	{ 0x100074c4, 0x37 },
-	{ 0x100074c5, 0xd7 },
-	{ 0x100074c6, 0x00 },
-	{ 0x100074c7, 0x00 },
-	{ 0x100074c8, 0x83 },
-	{ 0x100074c9, 0x47 },
-	{ 0x100074ca, 0x37 },
-	{ 0x100074cb, 0x54 },
-	{ 0x100074cc, 0x93 },
-	{ 0x100074cd, 0xf7 },
-	{ 0x100074ce, 0xf7 },
-	{ 0x100074cf, 0x0f },
-	{ 0x100074d0, 0x93 },
-	{ 0x100074d1, 0xe7 },
-	{ 0x100074d2, 0x07 },
-	{ 0x100074d3, 0x02 },
-	{ 0x100074d4, 0xa3 },
-	{ 0x100074d5, 0x01 },
-	{ 0x100074d6, 0xf7 },
-	{ 0x100074d7, 0x54 },
-	{ 0x100074d8, 0x83 },
-	{ 0x100074d9, 0x47 },
-	{ 0x100074da, 0x37 },
-	{ 0x100074db, 0x54 },
-	{ 0x100074dc, 0x93 },
-	{ 0x100074dd, 0xf7 },
-	{ 0x100074de, 0xf7 },
-	{ 0x100074df, 0x0d },
-	{ 0x100074e0, 0xa3 },
-	{ 0x100074e1, 0x01 },
-	{ 0x100074e2, 0xf7 },
-	{ 0x100074e3, 0x54 },
-	{ 0x100074e4, 0x23 },
-	{ 0x100074e5, 0xac },
-	{ 0x100074e6, 0x01 },
-	{ 0x100074e7, 0x56 },
-	{ 0x100074e8, 0x37 },
-	{ 0x100074e9, 0xd4 },
-	{ 0x100074ea, 0x00 },
-	{ 0x100074eb, 0x00 },
-	{ 0x100074ec, 0x83 },
-	{ 0x100074ed, 0x47 },
-	{ 0x100074ee, 0xd4 },
-	{ 0x100074ef, 0x47 },
-	{ 0x100074f0, 0x93 },
-	{ 0x100074f1, 0xf7 },
-	{ 0x100074f2, 0x17 },
-	{ 0x100074f3, 0x00 },
-	{ 0x100074f4, 0x63 },
-	{ 0x100074f5, 0x80 },
-	{ 0x100074f6, 0x07 },
-	{ 0x100074f7, 0x06 },
-	{ 0x100074f8, 0x37 },
-	{ 0x100074f9, 0xd7 },
-	{ 0x100074fa, 0x00 },
-	{ 0x100074fb, 0x10 },
-	{ 0x100074fc, 0x83 },
-	{ 0x100074fd, 0x47 },
-	{ 0x100074fe, 0x77 },
-	{ 0x100074ff, 0xd9 },
-	{ 0x10007500, 0x93 },
-	{ 0x10007501, 0x87 },
-	{ 0x10007502, 0x17 },
-	{ 0x10007503, 0x00 },
-	{ 0x10007504, 0x93 },
-	{ 0x10007505, 0xf7 },
-	{ 0x10007506, 0xf7 },
-	{ 0x10007507, 0x0f },
-	{ 0x10007508, 0xa3 },
-	{ 0x10007509, 0x0b },
-	{ 0x1000750a, 0xf7 },
-	{ 0x1000750b, 0xd8 },
-	{ 0x1000750c, 0x03 },
-	{ 0x1000750d, 0x47 },
-	{ 0x1000750e, 0x77 },
-	{ 0x1000750f, 0xd9 },
-	{ 0x10007510, 0x83 },
-	{ 0x10007511, 0x47 },
-	{ 0x10007512, 0xc4 },
-	{ 0x10007513, 0x47 },
-	{ 0x10007514, 0x13 },
-	{ 0x10007515, 0x77 },
-	{ 0x10007516, 0xf7 },
-	{ 0x10007517, 0x0f },
-	{ 0x10007518, 0x93 },
-	{ 0x10007519, 0xf7 },
-	{ 0x1000751a, 0xf7 },
-	{ 0x1000751b, 0x0f },
-	{ 0x1000751c, 0x63 },
-	{ 0x1000751d, 0x6c },
-	{ 0x1000751e, 0xf7 },
-	{ 0x1000751f, 0x02 },
-	{ 0x10007520, 0xb7 },
-	{ 0x10007521, 0xf4 },
-	{ 0x10007522, 0x00 },
-	{ 0x10007523, 0x00 },
-	{ 0x10007524, 0x93 },
-	{ 0x10007525, 0x05 },
-	{ 0x10007526, 0x00 },
-	{ 0x10007527, 0x01 },
-	{ 0x10007528, 0x13 },
-	{ 0x10007529, 0x85 },
-	{ 0x1000752a, 0x34 },
-	{ 0x1000752b, 0x52 },
-	{ 0x1000752c, 0xef },
-	{ 0x1000752d, 0xa0 },
-	{ 0x1000752e, 0x8f },
-	{ 0x1000752f, 0xc6 },
-	{ 0x10007530, 0x93 },
-	{ 0x10007531, 0x05 },
-	{ 0x10007532, 0x00 },
-	{ 0x10007533, 0x00 },
-	{ 0x10007534, 0x13 },
-	{ 0x10007535, 0x85 },
-	{ 0x10007536, 0x54 },
-	{ 0x10007537, 0x10 },
-	{ 0x10007538, 0xef },
-	{ 0x10007539, 0xa0 },
-	{ 0x1000753a, 0xcf },
-	{ 0x1000753b, 0xc5 },
-	{ 0x1000753c, 0x93 },
-	{ 0x1000753d, 0x05 },
-	{ 0x1000753e, 0x00 },
-	{ 0x1000753f, 0x00 },
-	{ 0x10007540, 0x13 },
-	{ 0x10007541, 0x85 },
-	{ 0x10007542, 0x74 },
-	{ 0x10007543, 0x10 },
-	{ 0x10007544, 0xef },
-	{ 0x10007545, 0xa0 },
-	{ 0x10007546, 0x0f },
-	{ 0x10007547, 0xc5 },
-	{ 0x10007548, 0x83 },
-	{ 0x10007549, 0x47 },
-	{ 0x1000754a, 0xd4 },
-	{ 0x1000754b, 0x47 },
-	{ 0x1000754c, 0x93 },
-	{ 0x1000754d, 0xf7 },
-	{ 0x1000754e, 0xe7 },
-	{ 0x1000754f, 0x0f },
-	{ 0x10007550, 0xa3 },
-	{ 0x10007551, 0x0e },
-	{ 0x10007552, 0xf4 },
-	{ 0x10007553, 0x46 },
-	{ 0x10007554, 0x83 },
-	{ 0x10007555, 0x20 },
-	{ 0x10007556, 0xc1 },
-	{ 0x10007557, 0x00 },
-	{ 0x10007558, 0x03 },
-	{ 0x10007559, 0x24 },
-	{ 0x1000755a, 0x81 },
-	{ 0x1000755b, 0x00 },
-	{ 0x1000755c, 0x83 },
-	{ 0x1000755d, 0x24 },
-	{ 0x1000755e, 0x41 },
-	{ 0x1000755f, 0x00 },
-	{ 0x10007560, 0x13 },
-	{ 0x10007561, 0x01 },
-	{ 0x10007562, 0x01 },
-	{ 0x10007563, 0x01 },
-	{ 0x10007564, 0x67 },
-	{ 0x10007565, 0x80 },
-	{ 0x10007566, 0x00 },
-	{ 0x10007567, 0x00 },
-	{ 0x10007568, 0x13 },
-	{ 0x10007569, 0x01 },
-	{ 0x1000756a, 0x01 },
-	{ 0x1000756b, 0xff },
-	{ 0x1000756c, 0x23 },
-	{ 0x1000756d, 0x24 },
-	{ 0x1000756e, 0x81 },
-	{ 0x1000756f, 0x00 },
-	{ 0x10007570, 0x23 },
-	{ 0x10007571, 0x26 },
-	{ 0x10007572, 0x11 },
-	{ 0x10007573, 0x00 },
-	{ 0x10007574, 0x23 },
-	{ 0x10007575, 0x22 },
-	{ 0x10007576, 0x91 },
-	{ 0x10007577, 0x00 },
-	{ 0x10007578, 0x37 },
-	{ 0x10007579, 0xd4 },
-	{ 0x1000757a, 0x00 },
-	{ 0x1000757b, 0x00 },
-	{ 0x1000757c, 0x83 },
-	{ 0x1000757d, 0x47 },
-	{ 0x1000757e, 0x04 },
-	{ 0x1000757f, 0x54 },
-	{ 0x10007580, 0x93 },
-	{ 0x10007581, 0x97 },
-	{ 0x10007582, 0x87 },
-	{ 0x10007583, 0x01 },
-	{ 0x10007584, 0x93 },
-	{ 0x10007585, 0xd7 },
-	{ 0x10007586, 0x87 },
-	{ 0x10007587, 0x41 },
-	{ 0x10007588, 0x63 },
-	{ 0x10007589, 0xd0 },
-	{ 0x1000758a, 0x07 },
-	{ 0x1000758b, 0x06 },
-	{ 0x1000758c, 0xb7 },
-	{ 0x1000758d, 0xf4 },
-	{ 0x1000758e, 0x00 },
-	{ 0x1000758f, 0x00 },
-	{ 0x10007590, 0x93 },
-	{ 0x10007591, 0x05 },
-	{ 0x10007592, 0x60 },
-	{ 0x10007593, 0x01 },
-	{ 0x10007594, 0x13 },
-	{ 0x10007595, 0x85 },
-	{ 0x10007596, 0x34 },
-	{ 0x10007597, 0x52 },
-	{ 0x10007598, 0xef },
-	{ 0x10007599, 0xa0 },
-	{ 0x1000759a, 0xcf },
-	{ 0x1000759b, 0xbf },
-	{ 0x1000759c, 0x93 },
-	{ 0x1000759d, 0x05 },
-	{ 0x1000759e, 0x00 },
-	{ 0x1000759f, 0x04 },
-	{ 0x100075a0, 0x13 },
-	{ 0x100075a1, 0x85 },
-	{ 0x100075a2, 0x54 },
-	{ 0x100075a3, 0x10 },
-	{ 0x100075a4, 0xef },
-	{ 0x100075a5, 0xa0 },
-	{ 0x100075a6, 0x0f },
-	{ 0x100075a7, 0xbf },
-	{ 0x100075a8, 0x93 },
-	{ 0x100075a9, 0x05 },
-	{ 0x100075aa, 0x00 },
-	{ 0x100075ab, 0x04 },
-	{ 0x100075ac, 0x13 },
-	{ 0x100075ad, 0x85 },
-	{ 0x100075ae, 0x74 },
-	{ 0x100075af, 0x10 },
-	{ 0x100075b0, 0xef },
-	{ 0x100075b1, 0xa0 },
-	{ 0x100075b2, 0x4f },
-	{ 0x100075b3, 0xbe },
-	{ 0x100075b4, 0x83 },
-	{ 0x100075b5, 0x47 },
-	{ 0x100075b6, 0xd4 },
-	{ 0x100075b7, 0x47 },
-	{ 0x100075b8, 0x37 },
-	{ 0x100075b9, 0xd7 },
-	{ 0x100075ba, 0x00 },
-	{ 0x100075bb, 0x10 },
-	{ 0x100075bc, 0x93 },
-	{ 0x100075bd, 0xf7 },
-	{ 0x100075be, 0xf7 },
-	{ 0x100075bf, 0x0f },
-	{ 0x100075c0, 0x93 },
-	{ 0x100075c1, 0xe7 },
-	{ 0x100075c2, 0x17 },
-	{ 0x100075c3, 0x00 },
-	{ 0x100075c4, 0xa3 },
-	{ 0x100075c5, 0x0e },
-	{ 0x100075c6, 0xf4 },
-	{ 0x100075c7, 0x46 },
-	{ 0x100075c8, 0xa3 },
-	{ 0x100075c9, 0x0b },
-	{ 0x100075ca, 0x07 },
-	{ 0x100075cb, 0xd8 },
-	{ 0x100075cc, 0x83 },
-	{ 0x100075cd, 0x47 },
-	{ 0x100075ce, 0x87 },
-	{ 0x100075cf, 0xd9 },
-	{ 0x100075d0, 0x93 },
-	{ 0x100075d1, 0x87 },
-	{ 0x100075d2, 0x17 },
-	{ 0x100075d3, 0x00 },
-	{ 0x100075d4, 0x93 },
-	{ 0x100075d5, 0xf7 },
-	{ 0x100075d6, 0xf7 },
-	{ 0x100075d7, 0x0f },
-	{ 0x100075d8, 0x23 },
-	{ 0x100075d9, 0x0c },
-	{ 0x100075da, 0xf7 },
-	{ 0x100075db, 0xd8 },
-	{ 0x100075dc, 0x83 },
-	{ 0x100075dd, 0x47 },
-	{ 0x100075de, 0x04 },
-	{ 0x100075df, 0x54 },
-	{ 0x100075e0, 0x93 },
-	{ 0x100075e1, 0xf7 },
-	{ 0x100075e2, 0xf7 },
-	{ 0x100075e3, 0x07 },
-	{ 0x100075e4, 0x23 },
-	{ 0x100075e5, 0x00 },
-	{ 0x100075e6, 0xf4 },
-	{ 0x100075e7, 0x54 },
-	{ 0x100075e8, 0x83 },
-	{ 0x100075e9, 0x20 },
-	{ 0x100075ea, 0xc1 },
-	{ 0x100075eb, 0x00 },
-	{ 0x100075ec, 0x03 },
-	{ 0x100075ed, 0x24 },
-	{ 0x100075ee, 0x81 },
-	{ 0x100075ef, 0x00 },
-	{ 0x100075f0, 0x83 },
-	{ 0x100075f1, 0x24 },
-	{ 0x100075f2, 0x41 },
-	{ 0x100075f3, 0x00 },
-	{ 0x100075f4, 0x13 },
-	{ 0x100075f5, 0x01 },
-	{ 0x100075f6, 0x01 },
-	{ 0x100075f7, 0x01 },
-	{ 0x100075f8, 0x67 },
-	{ 0x100075f9, 0x80 },
-	{ 0x100075fa, 0x00 },
-	{ 0x100075fb, 0x00 },
-	{ 0x100075fc, 0x13 },
-	{ 0x100075fd, 0x01 },
-	{ 0x100075fe, 0x01 },
-	{ 0x100075ff, 0xff },
-	{ 0x10007600, 0x23 },
-	{ 0x10007601, 0x24 },
-	{ 0x10007602, 0x81 },
-	{ 0x10007603, 0x00 },
-	{ 0x10007604, 0x37 },
-	{ 0x10007605, 0xd4 },
-	{ 0x10007606, 0x00 },
-	{ 0x10007607, 0x00 },
-	{ 0x10007608, 0x83 },
-	{ 0x10007609, 0x27 },
-	{ 0x1000760a, 0x04 },
-	{ 0x1000760b, 0x53 },
-	{ 0x1000760c, 0x23 },
-	{ 0x1000760d, 0x22 },
-	{ 0x1000760e, 0x91 },
-	{ 0x1000760f, 0x00 },
-	{ 0x10007610, 0xb7 },
-	{ 0x10007611, 0x04 },
-	{ 0x10007612, 0x00 },
-	{ 0x10007613, 0x40 },
-	{ 0x10007614, 0x23 },
-	{ 0x10007615, 0x26 },
-	{ 0x10007616, 0x11 },
-	{ 0x10007617, 0x00 },
-	{ 0x10007618, 0xb3 },
-	{ 0x10007619, 0xf7 },
-	{ 0x1000761a, 0x97 },
-	{ 0x1000761b, 0x00 },
-	{ 0x1000761c, 0x63 },
-	{ 0x1000761d, 0x86 },
-	{ 0x1000761e, 0x07 },
-	{ 0x1000761f, 0x00 },
-	{ 0x10007620, 0xef },
-	{ 0x10007621, 0xd0 },
-	{ 0x10007622, 0x5f },
-	{ 0x10007623, 0xc2 },
-	{ 0x10007624, 0x23 },
-	{ 0x10007625, 0x28 },
-	{ 0x10007626, 0x94 },
-	{ 0x10007627, 0x52 },
-	{ 0x10007628, 0x83 },
-	{ 0x10007629, 0x20 },
-	{ 0x1000762a, 0xc1 },
-	{ 0x1000762b, 0x00 },
-	{ 0x1000762c, 0x03 },
-	{ 0x1000762d, 0x24 },
-	{ 0x1000762e, 0x81 },
-	{ 0x1000762f, 0x00 },
-	{ 0x10007630, 0x83 },
-	{ 0x10007631, 0x24 },
-	{ 0x10007632, 0x41 },
-	{ 0x10007633, 0x00 },
-	{ 0x10007634, 0x13 },
-	{ 0x10007635, 0x01 },
-	{ 0x10007636, 0x01 },
-	{ 0x10007637, 0x01 },
-	{ 0x10007638, 0x67 },
-	{ 0x10007639, 0x80 },
-	{ 0x1000763a, 0x00 },
-	{ 0x1000763b, 0x00 },
-	{ 0x1000763c, 0x37 },
-	{ 0x1000763d, 0xc7 },
-	{ 0x1000763e, 0x00 },
-	{ 0x1000763f, 0x00 },
-	{ 0x10007640, 0x83 },
-	{ 0x10007641, 0x27 },
-	{ 0x10007642, 0xc7 },
-	{ 0x10007643, 0x5f },
-	{ 0x10007644, 0x23 },
-	{ 0x10007645, 0xa2 },
-	{ 0x10007646, 0xf1 },
-	{ 0x10007647, 0x42 },
-	{ 0x10007648, 0xb7 },
-	{ 0x10007649, 0x06 },
-	{ 0x1000764a, 0x00 },
-	{ 0x1000764b, 0x10 },
-	{ 0x1000764c, 0xb3 },
-	{ 0x1000764d, 0xf7 },
-	{ 0x1000764e, 0xd7 },
-	{ 0x1000764f, 0x00 },
-	{ 0x10007650, 0x63 },
-	{ 0x10007651, 0x80 },
-	{ 0x10007652, 0x07 },
-	{ 0x10007653, 0x0a },
-	{ 0x10007654, 0x83 },
-	{ 0x10007655, 0x47 },
-	{ 0x10007656, 0x07 },
-	{ 0x10007657, 0x56 },
-	{ 0x10007658, 0x93 },
-	{ 0x10007659, 0xf7 },
-	{ 0x1000765a, 0x87 },
-	{ 0x1000765b, 0x01 },
-	{ 0x1000765c, 0x63 },
-	{ 0x1000765d, 0x8a },
-	{ 0x1000765e, 0x07 },
-	{ 0x1000765f, 0x08 },
-	{ 0x10007660, 0x83 },
-	{ 0x10007661, 0x47 },
-	{ 0x10007662, 0x17 },
-	{ 0x10007663, 0x08 },
-	{ 0x10007664, 0x93 },
-	{ 0x10007665, 0xf7 },
-	{ 0x10007666, 0x47 },
-	{ 0x10007667, 0x00 },
-	{ 0x10007668, 0x63 },
-	{ 0x10007669, 0x84 },
-	{ 0x1000766a, 0x07 },
-	{ 0x1000766b, 0x08 },
-	{ 0x1000766c, 0x13 },
-	{ 0x1000766d, 0x01 },
-	{ 0x1000766e, 0x01 },
-	{ 0x1000766f, 0xff },
-	{ 0x10007670, 0x23 },
-	{ 0x10007671, 0x26 },
-	{ 0x10007672, 0x11 },
-	{ 0x10007673, 0x00 },
-	{ 0x10007674, 0xb7 },
-	{ 0x10007675, 0xc7 },
-	{ 0x10007676, 0xc2 },
-	{ 0x10007677, 0x3f },
-	{ 0x10007678, 0x03 },
-	{ 0x10007679, 0xa7 },
-	{ 0x1000767a, 0x07 },
-	{ 0x1000767b, 0xfc },
-	{ 0x1000767c, 0x63 },
-	{ 0x1000767d, 0x10 },
-	{ 0x1000767e, 0x05 },
-	{ 0x1000767f, 0x06 },
-	{ 0x10007680, 0x13 },
-	{ 0x10007681, 0x67 },
-	{ 0x10007682, 0x07 },
-	{ 0x10007683, 0x20 },
-	{ 0x10007684, 0x23 },
-	{ 0x10007685, 0xa0 },
-	{ 0x10007686, 0xe7 },
-	{ 0x10007687, 0xfc },
-	{ 0x10007688, 0x03 },
-	{ 0x10007689, 0xa7 },
-	{ 0x1000768a, 0x07 },
-	{ 0x1000768b, 0xfc },
-	{ 0x1000768c, 0x13 },
-	{ 0x1000768d, 0x67 },
-	{ 0x1000768e, 0x07 },
-	{ 0x1000768f, 0x40 },
-	{ 0x10007690, 0x23 },
-	{ 0x10007691, 0xa0 },
-	{ 0x10007692, 0xe7 },
-	{ 0x10007693, 0xfc },
-	{ 0x10007694, 0x37 },
-	{ 0x10007695, 0xc7 },
-	{ 0x10007696, 0xc2 },
-	{ 0x10007697, 0x3f },
-	{ 0x10007698, 0x83 },
-	{ 0x10007699, 0x27 },
-	{ 0x1000769a, 0x07 },
-	{ 0x1000769b, 0xfc },
-	{ 0x1000769c, 0x13 },
-	{ 0x1000769d, 0x75 },
-	{ 0x1000769e, 0x15 },
-	{ 0x1000769f, 0x00 },
-	{ 0x100076a0, 0x13 },
-	{ 0x100076a1, 0x15 },
-	{ 0x100076a2, 0x85 },
-	{ 0x100076a3, 0x00 },
-	{ 0x100076a4, 0x93 },
-	{ 0x100076a5, 0xf7 },
-	{ 0x100076a6, 0xf7 },
-	{ 0x100076a7, 0xef },
-	{ 0x100076a8, 0x33 },
-	{ 0x100076a9, 0xe5 },
-	{ 0x100076aa, 0xa7 },
-	{ 0x100076ab, 0x00 },
-	{ 0x100076ac, 0x23 },
-	{ 0x100076ad, 0x20 },
-	{ 0x100076ae, 0xa7 },
-	{ 0x100076af, 0xfc },
-	{ 0x100076b0, 0x93 },
-	{ 0x100076b1, 0x05 },
-	{ 0x100076b2, 0x00 },
-	{ 0x100076b3, 0x00 },
-	{ 0x100076b4, 0x13 },
-	{ 0x100076b5, 0x05 },
-	{ 0x100076b6, 0xa0 },
-	{ 0x100076b7, 0x00 },
-	{ 0x100076b8, 0xef },
-	{ 0x100076b9, 0xe0 },
-	{ 0x100076ba, 0xcf },
-	{ 0x100076bb, 0xb6 },
-	{ 0x100076bc, 0x37 },
-	{ 0x100076bd, 0xf7 },
-	{ 0x100076be, 0x00 },
-	{ 0x100076bf, 0x00 },
-	{ 0x100076c0, 0x83 },
-	{ 0x100076c1, 0x47 },
-	{ 0x100076c2, 0x57 },
-	{ 0x100076c3, 0x01 },
-	{ 0x100076c4, 0x93 },
-	{ 0x100076c5, 0xf7 },
-	{ 0x100076c6, 0xf7 },
-	{ 0x100076c7, 0x0f },
-	{ 0x100076c8, 0x93 },
-	{ 0x100076c9, 0xe7 },
-	{ 0x100076ca, 0x47 },
-	{ 0x100076cb, 0x00 },
-	{ 0x100076cc, 0xa3 },
-	{ 0x100076cd, 0x0a },
-	{ 0x100076ce, 0xf7 },
-	{ 0x100076cf, 0x00 },
-	{ 0x100076d0, 0x83 },
-	{ 0x100076d1, 0x20 },
-	{ 0x100076d2, 0xc1 },
-	{ 0x100076d3, 0x00 },
-	{ 0x100076d4, 0x13 },
-	{ 0x100076d5, 0x01 },
-	{ 0x100076d6, 0x01 },
-	{ 0x100076d7, 0x01 },
-	{ 0x100076d8, 0x67 },
-	{ 0x100076d9, 0x80 },
-	{ 0x100076da, 0x00 },
-	{ 0x100076db, 0x00 },
-	{ 0x100076dc, 0x13 },
-	{ 0x100076dd, 0x77 },
-	{ 0x100076de, 0xf7 },
-	{ 0x100076df, 0xdf },
-	{ 0x100076e0, 0x23 },
-	{ 0x100076e1, 0xa0 },
-	{ 0x100076e2, 0xe7 },
-	{ 0x100076e3, 0xfc },
-	{ 0x100076e4, 0x03 },
-	{ 0x100076e5, 0xa7 },
-	{ 0x100076e6, 0x07 },
-	{ 0x100076e7, 0xfc },
-	{ 0x100076e8, 0x13 },
-	{ 0x100076e9, 0x77 },
-	{ 0x100076ea, 0xf7 },
-	{ 0x100076eb, 0xbf },
-	{ 0x100076ec, 0x6f },
-	{ 0x100076ed, 0xf0 },
-	{ 0x100076ee, 0x5f },
-	{ 0x100076ef, 0xfa },
-	{ 0x100076f0, 0x67 },
-	{ 0x100076f1, 0x80 },
-	{ 0x100076f2, 0x00 },
-	{ 0x100076f3, 0x00 },
-	{ 0x100076f4, 0xb7 },
-	{ 0x100076f5, 0xc7 },
-	{ 0x100076f6, 0x00 },
-	{ 0x100076f7, 0x00 },
-	{ 0x100076f8, 0x03 },
-	{ 0x100076f9, 0xc7 },
-	{ 0x100076fa, 0x87 },
-	{ 0x100076fb, 0x59 },
-	{ 0x100076fc, 0x13 },
-	{ 0x100076fd, 0x77 },
-	{ 0x100076fe, 0xf7 },
-	{ 0x100076ff, 0x0f },
-	{ 0x10007700, 0x13 },
-	{ 0x10007701, 0x67 },
-	{ 0x10007702, 0x17 },
-	{ 0x10007703, 0x00 },
-	{ 0x10007704, 0x23 },
-	{ 0x10007705, 0x8c },
-	{ 0x10007706, 0xe7 },
-	{ 0x10007707, 0x58 },
-	{ 0x10007708, 0x03 },
-	{ 0x10007709, 0xc7 },
-	{ 0x1000770a, 0x77 },
-	{ 0x1000770b, 0x04 },
-	{ 0x1000770c, 0x13 },
-	{ 0x1000770d, 0x17 },
-	{ 0x1000770e, 0x87 },
-	{ 0x1000770f, 0x01 },
-	{ 0x10007710, 0x13 },
-	{ 0x10007711, 0x57 },
-	{ 0x10007712, 0x87 },
-	{ 0x10007713, 0x41 },
-	{ 0x10007714, 0x63 },
-	{ 0x10007715, 0x58 },
-	{ 0x10007716, 0x07 },
-	{ 0x10007717, 0x12 },
-	{ 0x10007718, 0x37 },
-	{ 0x10007719, 0xd7 },
-	{ 0x1000771a, 0x00 },
-	{ 0x1000771b, 0x00 },
-	{ 0x1000771c, 0x83 },
-	{ 0x1000771d, 0x26 },
-	{ 0x1000771e, 0x87 },
-	{ 0x1000771f, 0x53 },
-	{ 0x10007720, 0x37 },
-	{ 0x10007721, 0x06 },
-	{ 0x10007722, 0x00 },
-	{ 0x10007723, 0x40 },
-	{ 0x10007724, 0x93 },
-	{ 0x10007725, 0x05 },
-	{ 0x10007726, 0x80 },
-	{ 0x10007727, 0x01 },
-	{ 0x10007728, 0xb3 },
-	{ 0x10007729, 0xe6 },
-	{ 0x1000772a, 0xc6 },
-	{ 0x1000772b, 0x00 },
-	{ 0x1000772c, 0x23 },
-	{ 0x1000772d, 0x2c },
-	{ 0x1000772e, 0xd7 },
-	{ 0x1000772f, 0x52 },
-	{ 0x10007730, 0x83 },
-	{ 0x10007731, 0xc6 },
-	{ 0x10007732, 0x07 },
-	{ 0x10007733, 0x56 },
-	{ 0x10007734, 0x93 },
-	{ 0x10007735, 0xf6 },
-	{ 0x10007736, 0xf6 },
-	{ 0x10007737, 0x0f },
-	{ 0x10007738, 0x63 },
-	{ 0x10007739, 0x9c },
-	{ 0x1000773a, 0xb6 },
-	{ 0x1000773b, 0x0e },
-	{ 0x1000773c, 0x83 },
-	{ 0x1000773d, 0x27 },
-	{ 0x1000773e, 0x87 },
-	{ 0x1000773f, 0x53 },
-	{ 0x10007740, 0xb3 },
-	{ 0x10007741, 0xf7 },
-	{ 0x10007742, 0xc7 },
-	{ 0x10007743, 0x00 },
-	{ 0x10007744, 0x63 },
-	{ 0x10007745, 0x80 },
-	{ 0x10007746, 0x07 },
-	{ 0x10007747, 0x10 },
-	{ 0x10007748, 0x13 },
-	{ 0x10007749, 0x01 },
-	{ 0x1000774a, 0x01 },
-	{ 0x1000774b, 0xff },
-	{ 0x1000774c, 0x23 },
-	{ 0x1000774d, 0x24 },
-	{ 0x1000774e, 0x81 },
-	{ 0x1000774f, 0x00 },
-	{ 0x10007750, 0x83 },
-	{ 0x10007751, 0xa7 },
-	{ 0x10007752, 0x41 },
-	{ 0x10007753, 0x58 },
-	{ 0x10007754, 0x23 },
-	{ 0x10007755, 0x26 },
-	{ 0x10007756, 0x11 },
-	{ 0x10007757, 0x00 },
-	{ 0x10007758, 0x63 },
-	{ 0x10007759, 0x94 },
-	{ 0x1000775a, 0x07 },
-	{ 0x1000775b, 0x0c },
-	{ 0x1000775c, 0x83 },
-	{ 0x1000775d, 0x27 },
-	{ 0x1000775e, 0x07 },
-	{ 0x1000775f, 0x53 },
-	{ 0x10007760, 0x03 },
-	{ 0x10007761, 0xc6 },
-	{ 0x10007762, 0xb1 },
-	{ 0x10007763, 0x42 },
-	{ 0x10007764, 0x93 },
-	{ 0x10007765, 0xd7 },
-	{ 0x10007766, 0xe7 },
-	{ 0x10007767, 0x01 },
-	{ 0x10007768, 0x93 },
-	{ 0x10007769, 0xf7 },
-	{ 0x1000776a, 0x17 },
-	{ 0x1000776b, 0x00 },
-	{ 0x1000776c, 0x93 },
-	{ 0x1000776d, 0x06 },
-	{ 0x1000776e, 0x10 },
-	{ 0x1000776f, 0x00 },
-	{ 0x10007770, 0x63 },
-	{ 0x10007771, 0x14 },
-	{ 0x10007772, 0x06 },
-	{ 0x10007773, 0x00 },
-	{ 0x10007774, 0xb3 },
-	{ 0x10007775, 0x86 },
-	{ 0x10007776, 0xf6 },
-	{ 0x10007777, 0x40 },
-	{ 0x10007778, 0xa3 },
-	{ 0x10007779, 0x85 },
-	{ 0x1000777a, 0xd1 },
-	{ 0x1000777b, 0x42 },
-	{ 0x1000777c, 0x03 },
-	{ 0x1000777d, 0xc6 },
-	{ 0x1000777e, 0xa1 },
-	{ 0x1000777f, 0x42 },
-	{ 0x10007780, 0x93 },
-	{ 0x10007781, 0x06 },
-	{ 0x10007782, 0x10 },
-	{ 0x10007783, 0x00 },
-	{ 0x10007784, 0x63 },
-	{ 0x10007785, 0x14 },
-	{ 0x10007786, 0x06 },
-	{ 0x10007787, 0x00 },
-	{ 0x10007788, 0xb3 },
-	{ 0x10007789, 0x86 },
-	{ 0x1000778a, 0xf6 },
-	{ 0x1000778b, 0x40 },
-	{ 0x1000778c, 0x23 },
-	{ 0x1000778d, 0x85 },
-	{ 0x1000778e, 0xd1 },
-	{ 0x1000778f, 0x42 },
-	{ 0x10007790, 0x03 },
-	{ 0x10007791, 0xc6 },
-	{ 0x10007792, 0x91 },
-	{ 0x10007793, 0x42 },
-	{ 0x10007794, 0x93 },
-	{ 0x10007795, 0x06 },
-	{ 0x10007796, 0x10 },
-	{ 0x10007797, 0x00 },
-	{ 0x10007798, 0x63 },
-	{ 0x10007799, 0x14 },
-	{ 0x1000779a, 0x06 },
-	{ 0x1000779b, 0x00 },
-	{ 0x1000779c, 0xb3 },
-	{ 0x1000779d, 0x86 },
-	{ 0x1000779e, 0xf6 },
-	{ 0x1000779f, 0x40 },
-	{ 0x100077a0, 0xa3 },
-	{ 0x100077a1, 0x84 },
-	{ 0x100077a2, 0xd1 },
-	{ 0x100077a3, 0x42 },
-	{ 0x100077a4, 0x03 },
-	{ 0x100077a5, 0xc6 },
-	{ 0x100077a6, 0x81 },
-	{ 0x100077a7, 0x42 },
-	{ 0x100077a8, 0x93 },
-	{ 0x100077a9, 0x06 },
-	{ 0x100077aa, 0x10 },
-	{ 0x100077ab, 0x00 },
-	{ 0x100077ac, 0x63 },
-	{ 0x100077ad, 0x14 },
-	{ 0x100077ae, 0x06 },
-	{ 0x100077af, 0x00 },
-	{ 0x100077b0, 0xb3 },
-	{ 0x100077b1, 0x86 },
-	{ 0x100077b2, 0xf6 },
-	{ 0x100077b3, 0x40 },
-	{ 0x100077b4, 0x23 },
-	{ 0x100077b5, 0x84 },
-	{ 0x100077b6, 0xd1 },
-	{ 0x100077b7, 0x42 },
-	{ 0x100077b8, 0xb7 },
-	{ 0x100077b9, 0xd7 },
-	{ 0x100077ba, 0x00 },
-	{ 0x100077bb, 0x00 },
-	{ 0x100077bc, 0x83 },
-	{ 0x100077bd, 0xa7 },
-	{ 0x100077be, 0x07 },
-	{ 0x100077bf, 0x53 },
-	{ 0x100077c0, 0x37 },
-	{ 0x100077c1, 0x07 },
-	{ 0x100077c2, 0x00 },
-	{ 0x100077c3, 0x40 },
-	{ 0x100077c4, 0xb3 },
-	{ 0x100077c5, 0xf7 },
-	{ 0x100077c6, 0xe7 },
-	{ 0x100077c7, 0x00 },
-	{ 0x100077c8, 0x63 },
-	{ 0x100077c9, 0x8c },
-	{ 0x100077ca, 0x07 },
-	{ 0x100077cb, 0x04 },
-	{ 0x100077cc, 0xb7 },
-	{ 0x100077cd, 0x47 },
-	{ 0x100077ce, 0x0f },
-	{ 0x100077cf, 0x00 },
-	{ 0x100077d0, 0x93 },
-	{ 0x100077d1, 0x87 },
-	{ 0x100077d2, 0x17 },
-	{ 0x100077d3, 0x24 },
-	{ 0x100077d4, 0xb7 },
-	{ 0x100077d5, 0xf6 },
-	{ 0x100077d6, 0x00 },
-	{ 0x100077d7, 0x00 },
-	{ 0x100077d8, 0x03 },
-	{ 0x100077d9, 0xc7 },
-	{ 0x100077da, 0xf6 },
-	{ 0x100077db, 0x83 },
-	{ 0x100077dc, 0x13 },
-	{ 0x100077dd, 0x77 },
-	{ 0x100077de, 0x07 },
-	{ 0x100077df, 0x04 },
-	{ 0x100077e0, 0x63 },
-	{ 0x100077e1, 0x16 },
-	{ 0x100077e2, 0x07 },
-	{ 0x100077e3, 0x00 },
-	{ 0x100077e4, 0x93 },
-	{ 0x100077e5, 0x87 },
-	{ 0x100077e6, 0xf7 },
-	{ 0x100077e7, 0xff },
-	{ 0x100077e8, 0xe3 },
-	{ 0x100077e9, 0x98 },
-	{ 0x100077ea, 0x07 },
-	{ 0x100077eb, 0xfe },
-	{ 0x100077ec, 0x13 },
-	{ 0x100077ed, 0x05 },
-	{ 0x100077ee, 0x80 },
-	{ 0x100077ef, 0x3e },
-	{ 0x100077f0, 0x93 },
-	{ 0x100077f1, 0x05 },
-	{ 0x100077f2, 0x00 },
-	{ 0x100077f3, 0x00 },
-	{ 0x100077f4, 0xef },
-	{ 0x100077f5, 0xe0 },
-	{ 0x100077f6, 0x0f },
-	{ 0x100077f7, 0xa3 },
-	{ 0x100077f8, 0x37 },
-	{ 0x100077f9, 0xf7 },
-	{ 0x100077fa, 0x00 },
-	{ 0x100077fb, 0x00 },
-	{ 0x100077fc, 0x83 },
-	{ 0x100077fd, 0x47 },
-	{ 0x100077fe, 0xb7 },
-	{ 0x100077ff, 0x80 },
-	{ 0x10007800, 0x93 },
-	{ 0x10007801, 0xe7 },
-	{ 0x10007802, 0x07 },
-	{ 0x10007803, 0xf8 },
-	{ 0x10007804, 0x93 },
-	{ 0x10007805, 0xf7 },
-	{ 0x10007806, 0xf7 },
-	{ 0x10007807, 0x0f },
-	{ 0x10007808, 0xa3 },
-	{ 0x10007809, 0x05 },
-	{ 0x1000780a, 0xf7 },
-	{ 0x1000780b, 0x80 },
-	{ 0x1000780c, 0xb7 },
-	{ 0x1000780d, 0xd7 },
-	{ 0x1000780e, 0x00 },
-	{ 0x1000780f, 0x00 },
-	{ 0x10007810, 0x37 },
-	{ 0x10007811, 0x07 },
-	{ 0x10007812, 0x00 },
-	{ 0x10007813, 0x40 },
-	{ 0x10007814, 0x23 },
-	{ 0x10007815, 0xa8 },
-	{ 0x10007816, 0xe7 },
-	{ 0x10007817, 0x52 },
-	{ 0x10007818, 0x93 },
-	{ 0x10007819, 0x07 },
-	{ 0x1000781a, 0x10 },
-	{ 0x1000781b, 0x00 },
-	{ 0x1000781c, 0x23 },
-	{ 0x1000781d, 0xa2 },
-	{ 0x1000781e, 0xf1 },
-	{ 0x1000781f, 0x58 },
-	{ 0x10007820, 0x83 },
-	{ 0x10007821, 0x20 },
-	{ 0x10007822, 0xc1 },
-	{ 0x10007823, 0x00 },
-	{ 0x10007824, 0x03 },
-	{ 0x10007825, 0x24 },
-	{ 0x10007826, 0x81 },
-	{ 0x10007827, 0x00 },
-	{ 0x10007828, 0x13 },
-	{ 0x10007829, 0x01 },
-	{ 0x1000782a, 0x01 },
-	{ 0x1000782b, 0x01 },
-	{ 0x1000782c, 0x67 },
-	{ 0x1000782d, 0x80 },
-	{ 0x1000782e, 0x00 },
-	{ 0x1000782f, 0x00 },
-	{ 0x10007830, 0x83 },
-	{ 0x10007831, 0xc7 },
-	{ 0x10007832, 0x07 },
-	{ 0x10007833, 0x56 },
-	{ 0x10007834, 0x93 },
-	{ 0x10007835, 0xf7 },
-	{ 0x10007836, 0xf7 },
-	{ 0x10007837, 0x0f },
-	{ 0x10007838, 0x63 },
-	{ 0x10007839, 0x96 },
-	{ 0x1000783a, 0x07 },
-	{ 0x1000783b, 0x00 },
-	{ 0x1000783c, 0x23 },
-	{ 0x1000783d, 0xa2 },
-	{ 0x1000783e, 0x01 },
-	{ 0x1000783f, 0x58 },
-	{ 0x10007840, 0x67 },
-	{ 0x10007841, 0x80 },
-	{ 0x10007842, 0x00 },
-	{ 0x10007843, 0x00 },
-	{ 0x10007844, 0x67 },
-	{ 0x10007845, 0x80 },
-	{ 0x10007846, 0x00 },
-	{ 0x10007847, 0x00 },
-	{ 0x10007848, 0xb7 },
-	{ 0x10007849, 0xc7 },
-	{ 0x1000784a, 0x00 },
-	{ 0x1000784b, 0x00 },
-	{ 0x1000784c, 0x83 },
-	{ 0x1000784d, 0xc7 },
-	{ 0x1000784e, 0x07 },
-	{ 0x1000784f, 0x56 },
-	{ 0x10007850, 0x13 },
-	{ 0x10007851, 0x07 },
-	{ 0x10007852, 0x80 },
-	{ 0x10007853, 0x01 },
-	{ 0x10007854, 0x93 },
-	{ 0x10007855, 0xf7 },
-	{ 0x10007856, 0xf7 },
-	{ 0x10007857, 0x0f },
-	{ 0x10007858, 0x63 },
-	{ 0x10007859, 0x98 },
-	{ 0x1000785a, 0xe7 },
-	{ 0x1000785b, 0x00 },
-	{ 0x1000785c, 0x13 },
-	{ 0x1000785d, 0x05 },
-	{ 0x1000785e, 0x00 },
-	{ 0x1000785f, 0x7d },
-	{ 0x10007860, 0x93 },
-	{ 0x10007861, 0x05 },
-	{ 0x10007862, 0x00 },
-	{ 0x10007863, 0x00 },
-	{ 0x10007864, 0x6f },
-	{ 0x10007865, 0xe0 },
-	{ 0x10007866, 0x0f },
-	{ 0x10007867, 0x9c },
-	{ 0x10007868, 0x67 },
-	{ 0x10007869, 0x80 },
-	{ 0x1000786a, 0x00 },
-	{ 0x1000786b, 0x00 },
-	{ 0x1000786c, 0x13 },
-	{ 0x1000786d, 0x01 },
-	{ 0x1000786e, 0x01 },
-	{ 0x1000786f, 0xff },
-	{ 0x10007870, 0x23 },
-	{ 0x10007871, 0x26 },
-	{ 0x10007872, 0x11 },
-	{ 0x10007873, 0x00 },
-	{ 0x10007874, 0x23 },
-	{ 0x10007875, 0x24 },
-	{ 0x10007876, 0x81 },
-	{ 0x10007877, 0x00 },
-	{ 0x10007878, 0xef },
-	{ 0x10007879, 0xd0 },
-	{ 0x1000787a, 0x4f },
-	{ 0x1000787b, 0x91 },
-	{ 0x1000787c, 0x83 },
-	{ 0x1000787d, 0xc7 },
-	{ 0x1000787e, 0x81 },
-	{ 0x1000787f, 0x41 },
-	{ 0x10007880, 0x63 },
-	{ 0x10007881, 0x84 },
-	{ 0x10007882, 0x07 },
-	{ 0x10007883, 0x08 },
-	{ 0x10007884, 0xb7 },
-	{ 0x10007885, 0xd7 },
-	{ 0x10007886, 0x00 },
-	{ 0x10007887, 0x00 },
-	{ 0x10007888, 0x83 },
-	{ 0x10007889, 0xc7 },
-	{ 0x1000788a, 0x07 },
-	{ 0x1000788b, 0x47 },
-	{ 0x1000788c, 0x93 },
-	{ 0x1000788d, 0xf7 },
-	{ 0x1000788e, 0x07 },
-	{ 0x1000788f, 0x02 },
-	{ 0x10007890, 0x63 },
-	{ 0x10007891, 0x8a },
-	{ 0x10007892, 0x07 },
-	{ 0x10007893, 0x04 },
-	{ 0x10007894, 0x83 },
-	{ 0x10007895, 0xc7 },
-	{ 0x10007896, 0x11 },
-	{ 0x10007897, 0x44 },
-	{ 0x10007898, 0x93 },
-	{ 0x10007899, 0xf7 },
-	{ 0x1000789a, 0xd7 },
-	{ 0x1000789b, 0x0f },
-	{ 0x1000789c, 0x63 },
-	{ 0x1000789d, 0x90 },
-	{ 0x1000789e, 0x07 },
-	{ 0x1000789f, 0x02 },
-	{ 0x100078a0, 0x03 },
-	{ 0x100078a1, 0xc7 },
-	{ 0x100078a2, 0xd1 },
-	{ 0x100078a3, 0x58 },
-	{ 0x100078a4, 0xb7 },
-	{ 0x100078a5, 0x07 },
-	{ 0x100078a6, 0x00 },
-	{ 0x100078a7, 0x11 },
-	{ 0x100078a8, 0x23 },
-	{ 0x100078a9, 0x88 },
-	{ 0x100078aa, 0xe7 },
-	{ 0x100078ab, 0x00 },
-	{ 0x100078ac, 0x23 },
-	{ 0x100078ad, 0x88 },
-	{ 0x100078ae, 0xe7 },
-	{ 0x100078af, 0x20 },
-	{ 0x100078b0, 0x03 },
-	{ 0x100078b1, 0xc7 },
-	{ 0x100078b2, 0xc1 },
-	{ 0x100078b3, 0x58 },
-	{ 0x100078b4, 0x23 },
-	{ 0x100078b5, 0x8c },
-	{ 0x100078b6, 0xe7 },
-	{ 0x100078b7, 0x00 },
-	{ 0x100078b8, 0x6f },
-	{ 0x100078b9, 0x00 },
-	{ 0x100078ba, 0x80 },
-	{ 0x100078bb, 0x04 },
-	{ 0x100078bc, 0xb7 },
-	{ 0x100078bd, 0x07 },
-	{ 0x100078be, 0x00 },
-	{ 0x100078bf, 0x11 },
-	{ 0x100078c0, 0x23 },
-	{ 0x100078c1, 0x88 },
-	{ 0x100078c2, 0x07 },
-	{ 0x100078c3, 0x00 },
-	{ 0x100078c4, 0x23 },
-	{ 0x100078c5, 0x88 },
-	{ 0x100078c6, 0x07 },
-	{ 0x100078c7, 0x20 },
-	{ 0x100078c8, 0x23 },
-	{ 0x100078c9, 0x8c },
-	{ 0x100078ca, 0x07 },
-	{ 0x100078cb, 0x00 },
-	{ 0x100078cc, 0x23 },
-	{ 0x100078cd, 0x8c },
-	{ 0x100078ce, 0x07 },
-	{ 0x100078cf, 0x20 },
-	{ 0x100078d0, 0xef },
-	{ 0x100078d1, 0xb0 },
-	{ 0x100078d2, 0xcf },
-	{ 0x100078d3, 0xc4 },
-	{ 0x100078d4, 0x03 },
-	{ 0x100078d5, 0x24 },
-	{ 0x100078d6, 0x81 },
-	{ 0x100078d7, 0x00 },
-	{ 0x100078d8, 0x83 },
-	{ 0x100078d9, 0x20 },
-	{ 0x100078da, 0xc1 },
-	{ 0x100078db, 0x00 },
-	{ 0x100078dc, 0x13 },
-	{ 0x100078dd, 0x01 },
-	{ 0x100078de, 0x01 },
-	{ 0x100078df, 0x01 },
-	{ 0x100078e0, 0x6f },
-	{ 0x100078e1, 0xb0 },
-	{ 0x100078e2, 0xcf },
-	{ 0x100078e3, 0xcd },
-	{ 0x100078e4, 0x03 },
-	{ 0x100078e5, 0xc7 },
-	{ 0x100078e6, 0xd1 },
-	{ 0x100078e7, 0x58 },
-	{ 0x100078e8, 0xb7 },
-	{ 0x100078e9, 0x07 },
-	{ 0x100078ea, 0x00 },
-	{ 0x100078eb, 0x11 },
-	{ 0x100078ec, 0x23 },
-	{ 0x100078ed, 0x88 },
-	{ 0x100078ee, 0xe7 },
-	{ 0x100078ef, 0x00 },
-	{ 0x100078f0, 0x83 },
-	{ 0x100078f1, 0xc6 },
-	{ 0x100078f2, 0xc1 },
-	{ 0x100078f3, 0x58 },
-	{ 0x100078f4, 0x23 },
-	{ 0x100078f5, 0x88 },
-	{ 0x100078f6, 0xd7 },
-	{ 0x100078f7, 0x20 },
-	{ 0x100078f8, 0x23 },
-	{ 0x100078f9, 0x8c },
-	{ 0x100078fa, 0xd7 },
-	{ 0x100078fb, 0x00 },
-	{ 0x100078fc, 0x03 },
-	{ 0x100078fd, 0xc7 },
-	{ 0x100078fe, 0xc1 },
-	{ 0x100078ff, 0x58 },
-	{ 0x10007900, 0x23 },
-	{ 0x10007901, 0x8c },
-	{ 0x10007902, 0xe7 },
-	{ 0x10007903, 0x20 },
-	{ 0x10007904, 0x6f },
-	{ 0x10007905, 0xf0 },
-	{ 0x10007906, 0xdf },
-	{ 0x10007907, 0xfc },
-	{ 0x10007908, 0xb7 },
-	{ 0x10007909, 0x06 },
-	{ 0x1000790a, 0x00 },
-	{ 0x1000790b, 0x11 },
-	{ 0x1000790c, 0x03 },
-	{ 0x1000790d, 0xc7 },
-	{ 0x1000790e, 0x06 },
-	{ 0x1000790f, 0x21 },
-	{ 0x10007910, 0x03 },
-	{ 0x10007911, 0xc6 },
-	{ 0x10007912, 0xd1 },
-	{ 0x10007913, 0x58 },
-	{ 0x10007914, 0x13 },
-	{ 0x10007915, 0x84 },
-	{ 0x10007916, 0x07 },
-	{ 0x10007917, 0x00 },
-	{ 0x10007918, 0x13 },
-	{ 0x10007919, 0x77 },
-	{ 0x1000791a, 0xf7 },
-	{ 0x1000791b, 0x0f },
-	{ 0x1000791c, 0x63 },
-	{ 0x1000791d, 0x1a },
-	{ 0x1000791e, 0xe6 },
-	{ 0x1000791f, 0x00 },
-	{ 0x10007920, 0x83 },
-	{ 0x10007921, 0xc7 },
-	{ 0x10007922, 0x86 },
-	{ 0x10007923, 0x21 },
-	{ 0x10007924, 0x03 },
-	{ 0x10007925, 0xc7 },
-	{ 0x10007926, 0xc1 },
-	{ 0x10007927, 0x58 },
-	{ 0x10007928, 0x93 },
-	{ 0x10007929, 0xf7 },
-	{ 0x1000792a, 0xf7 },
-	{ 0x1000792b, 0x0f },
-	{ 0x1000792c, 0xe3 },
-	{ 0x1000792d, 0x02 },
-	{ 0x1000792e, 0xf7 },
-	{ 0x1000792f, 0xfa },
-	{ 0x10007930, 0xb7 },
-	{ 0x10007931, 0xc7 },
-	{ 0x10007932, 0x00 },
-	{ 0x10007933, 0x00 },
-	{ 0x10007934, 0x83 },
-	{ 0x10007935, 0xc7 },
-	{ 0x10007936, 0x07 },
-	{ 0x10007937, 0x56 },
-	{ 0x10007938, 0x13 },
-	{ 0x10007939, 0x07 },
-	{ 0x1000793a, 0xf0 },
-	{ 0x1000793b, 0x00 },
-	{ 0x1000793c, 0x93 },
-	{ 0x1000793d, 0xf7 },
-	{ 0x1000793e, 0xf7 },
-	{ 0x1000793f, 0x0f },
-	{ 0x10007940, 0xe3 },
-	{ 0x10007941, 0x78 },
-	{ 0x10007942, 0xf7 },
-	{ 0x10007943, 0xf8 },
-	{ 0x10007944, 0xb7 },
-	{ 0x10007945, 0xd7 },
-	{ 0x10007946, 0x00 },
-	{ 0x10007947, 0x00 },
-	{ 0x10007948, 0x83 },
-	{ 0x10007949, 0xc5 },
-	{ 0x1000794a, 0xa7 },
-	{ 0x1000794b, 0x47 },
-	{ 0x1000794c, 0x93 },
-	{ 0x1000794d, 0xf7 },
-	{ 0x1000794e, 0xf5 },
-	{ 0x1000794f, 0x0f },
-	{ 0x10007950, 0x93 },
-	{ 0x10007951, 0x95 },
-	{ 0x10007952, 0x57 },
-	{ 0x10007953, 0x00 },
-	{ 0x10007954, 0xb3 },
-	{ 0x10007955, 0x85 },
-	{ 0x10007956, 0xf5 },
-	{ 0x10007957, 0x40 },
-	{ 0x10007958, 0x93 },
-	{ 0x10007959, 0x95 },
-	{ 0x1000795a, 0x25 },
-	{ 0x1000795b, 0x00 },
-	{ 0x1000795c, 0xb3 },
-	{ 0x1000795d, 0x85 },
-	{ 0x1000795e, 0xf5 },
-	{ 0x1000795f, 0x00 },
-	{ 0x10007960, 0x13 },
-	{ 0x10007961, 0x95 },
-	{ 0x10007962, 0x35 },
-	{ 0x10007963, 0x00 },
-	{ 0x10007964, 0x93 },
-	{ 0x10007965, 0xd5 },
-	{ 0x10007966, 0xf5 },
-	{ 0x10007967, 0x41 },
-	{ 0x10007968, 0xef },
-	{ 0x10007969, 0xe0 },
-	{ 0x1000796a, 0xcf },
-	{ 0x1000796b, 0x8b },
-	{ 0x1000796c, 0x03 },
-	{ 0x1000796d, 0xc7 },
-	{ 0x1000796e, 0xd1 },
-	{ 0x1000796f, 0x58 },
-	{ 0x10007970, 0x6f },
-	{ 0x10007971, 0xf0 },
-	{ 0x10007972, 0x5f },
-	{ 0x10007973, 0xf3 },
-	{ 0x10007974, 0x13 },
-	{ 0x10007975, 0x01 },
-	{ 0x10007976, 0x01 },
-	{ 0x10007977, 0xfe },
-	{ 0x10007978, 0x23 },
-	{ 0x10007979, 0x2c },
-	{ 0x1000797a, 0x81 },
-	{ 0x1000797b, 0x00 },
-	{ 0x1000797c, 0x83 },
-	{ 0x1000797d, 0xc7 },
-	{ 0x1000797e, 0x21 },
-	{ 0x1000797f, 0x41 },
-	{ 0x10007980, 0x23 },
-	{ 0x10007981, 0x2e },
-	{ 0x10007982, 0x11 },
-	{ 0x10007983, 0x00 },
-	{ 0x10007984, 0x23 },
-	{ 0x10007985, 0x2a },
-	{ 0x10007986, 0x91 },
-	{ 0x10007987, 0x00 },
-	{ 0x10007988, 0x23 },
-	{ 0x10007989, 0x28 },
-	{ 0x1000798a, 0x21 },
-	{ 0x1000798b, 0x01 },
-	{ 0x1000798c, 0x23 },
-	{ 0x1000798d, 0x26 },
-	{ 0x1000798e, 0x31 },
-	{ 0x1000798f, 0x01 },
-	{ 0x10007990, 0x13 },
-	{ 0x10007991, 0x07 },
-	{ 0x10007992, 0x10 },
-	{ 0x10007993, 0x00 },
-	{ 0x10007994, 0x63 },
-	{ 0x10007995, 0x92 },
-	{ 0x10007996, 0xe7 },
-	{ 0x10007997, 0x02 },
-	{ 0x10007998, 0xa3 },
-	{ 0x10007999, 0x81 },
-	{ 0x1000799a, 0xf1 },
-	{ 0x1000799b, 0x40 },
-	{ 0x1000799c, 0x83 },
-	{ 0x1000799d, 0x20 },
-	{ 0x1000799e, 0xc1 },
-	{ 0x1000799f, 0x01 },
-	{ 0x100079a0, 0x03 },
-	{ 0x100079a1, 0x24 },
-	{ 0x100079a2, 0x81 },
-	{ 0x100079a3, 0x01 },
-	{ 0x100079a4, 0x83 },
-	{ 0x100079a5, 0x24 },
-	{ 0x100079a6, 0x41 },
-	{ 0x100079a7, 0x01 },
-	{ 0x100079a8, 0x03 },
-	{ 0x100079a9, 0x29 },
-	{ 0x100079aa, 0x01 },
-	{ 0x100079ab, 0x01 },
-	{ 0x100079ac, 0x83 },
-	{ 0x100079ad, 0x29 },
-	{ 0x100079ae, 0xc1 },
-	{ 0x100079af, 0x00 },
-	{ 0x100079b0, 0x13 },
-	{ 0x100079b1, 0x01 },
-	{ 0x100079b2, 0x01 },
-	{ 0x100079b3, 0x02 },
-	{ 0x100079b4, 0x67 },
-	{ 0x100079b5, 0x80 },
-	{ 0x100079b6, 0x00 },
-	{ 0x100079b7, 0x00 },
-	{ 0x100079b8, 0xe3 },
-	{ 0x100079b9, 0x92 },
-	{ 0x100079ba, 0x07 },
-	{ 0x100079bb, 0xfe },
-	{ 0x100079bc, 0x37 },
-	{ 0x100079bd, 0xc9 },
-	{ 0x100079be, 0x00 },
-	{ 0x100079bf, 0x00 },
-	{ 0x100079c0, 0x83 },
-	{ 0x100079c1, 0x47 },
-	{ 0x100079c2, 0x09 },
-	{ 0x100079c3, 0x56 },
-	{ 0x100079c4, 0x13 },
-	{ 0x100079c5, 0x07 },
-	{ 0x100079c6, 0x80 },
-	{ 0x100079c7, 0x01 },
-	{ 0x100079c8, 0x93 },
-	{ 0x100079c9, 0xf7 },
-	{ 0x100079ca, 0xf7 },
-	{ 0x100079cb, 0x0f },
-	{ 0x100079cc, 0xe3 },
-	{ 0x100079cd, 0x78 },
-	{ 0x100079ce, 0xf7 },
-	{ 0x100079cf, 0xfc },
-	{ 0x100079d0, 0x83 },
-	{ 0x100079d1, 0xc7 },
-	{ 0x100079d2, 0x31 },
-	{ 0x100079d3, 0x40 },
-	{ 0x100079d4, 0xe3 },
-	{ 0x100079d5, 0x84 },
-	{ 0x100079d6, 0x07 },
-	{ 0x100079d7, 0xfc },
-	{ 0x100079d8, 0xb7 },
-	{ 0x100079d9, 0xd4 },
-	{ 0x100079da, 0x00 },
-	{ 0x100079db, 0x00 },
-	{ 0x100079dc, 0x03 },
-	{ 0x100079dd, 0xc5 },
-	{ 0x100079de, 0x94 },
-	{ 0x100079df, 0x47 },
-	{ 0x100079e0, 0xb7 },
-	{ 0x100079e1, 0x15 },
-	{ 0x100079e2, 0x00 },
-	{ 0x100079e3, 0x00 },
-	{ 0x100079e4, 0x93 },
-	{ 0x100079e5, 0x85 },
-	{ 0x100079e6, 0x85 },
-	{ 0x100079e7, 0x38 },
-	{ 0x100079e8, 0x13 },
-	{ 0x100079e9, 0x75 },
-	{ 0x100079ea, 0xf5 },
-	{ 0x100079eb, 0x0f },
-	{ 0x100079ec, 0xef },
-	{ 0x100079ed, 0xe0 },
-	{ 0x100079ee, 0x5f },
-	{ 0x100079ef, 0xe0 },
-	{ 0x100079f0, 0x93 },
-	{ 0x100079f1, 0x55 },
-	{ 0x100079f2, 0xf5 },
-	{ 0x100079f3, 0x41 },
-	{ 0x100079f4, 0xef },
-	{ 0x100079f5, 0xe0 },
-	{ 0x100079f6, 0x0f },
-	{ 0x100079f7, 0x83 },
-	{ 0x100079f8, 0xa3 },
-	{ 0x100079f9, 0x81 },
-	{ 0x100079fa, 0x01 },
-	{ 0x100079fb, 0x40 },
-	{ 0x100079fc, 0x83 },
-	{ 0x100079fd, 0x27 },
-	{ 0x100079fe, 0xc9 },
-	{ 0x100079ff, 0x5f },
-	{ 0x10007a00, 0x37 },
-	{ 0x10007a01, 0x07 },
-	{ 0x10007a02, 0x00 },
-	{ 0x10007a03, 0x02 },
-	{ 0x10007a04, 0xb3 },
-	{ 0x10007a05, 0xf7 },
-	{ 0x10007a06, 0xe7 },
-	{ 0x10007a07, 0x00 },
-	{ 0x10007a08, 0xe3 },
-	{ 0x10007a09, 0x8a },
-	{ 0x10007a0a, 0x07 },
-	{ 0x10007a0b, 0xf8 },
-	{ 0x10007a0c, 0x03 },
-	{ 0x10007a0d, 0xc7 },
-	{ 0x10007a0e, 0x04 },
-	{ 0x10007a0f, 0x90 },
-	{ 0x10007a10, 0x93 },
-	{ 0x10007a11, 0x07 },
-	{ 0x10007a12, 0x10 },
-	{ 0x10007a13, 0x00 },
-	{ 0x10007a14, 0x13 },
-	{ 0x10007a15, 0x77 },
-	{ 0x10007a16, 0x17 },
-	{ 0x10007a17, 0x00 },
-	{ 0x10007a18, 0x63 },
-	{ 0x10007a19, 0x1c },
-	{ 0x10007a1a, 0x07 },
-	{ 0x10007a1b, 0x00 },
-	{ 0x10007a1c, 0x83 },
-	{ 0x10007a1d, 0xc7 },
-	{ 0x10007a1e, 0x34 },
-	{ 0x10007a1f, 0x54 },
-	{ 0x10007a20, 0x93 },
-	{ 0x10007a21, 0xf7 },
-	{ 0x10007a22, 0xf7 },
-	{ 0x10007a23, 0x0f },
-	{ 0x10007a24, 0x93 },
-	{ 0x10007a25, 0xd7 },
-	{ 0x10007a26, 0x17 },
-	{ 0x10007a27, 0x00 },
-	{ 0x10007a28, 0x93 },
-	{ 0x10007a29, 0xc7 },
-	{ 0x10007a2a, 0x17 },
-	{ 0x10007a2b, 0x00 },
-	{ 0x10007a2c, 0x93 },
-	{ 0x10007a2d, 0xf7 },
-	{ 0x10007a2e, 0x17 },
-	{ 0x10007a2f, 0x00 },
-	{ 0x10007a30, 0xa3 },
-	{ 0x10007a31, 0x85 },
-	{ 0x10007a32, 0xf1 },
-	{ 0x10007a33, 0x42 },
-	{ 0x10007a34, 0x37 },
-	{ 0x10007a35, 0xd6 },
-	{ 0x10007a36, 0x00 },
-	{ 0x10007a37, 0x00 },
-	{ 0x10007a38, 0x03 },
-	{ 0x10007a39, 0x47 },
-	{ 0x10007a3a, 0x06 },
-	{ 0x10007a3b, 0x90 },
-	{ 0x10007a3c, 0x93 },
-	{ 0x10007a3d, 0x06 },
-	{ 0x10007a3e, 0x10 },
-	{ 0x10007a3f, 0x00 },
-	{ 0x10007a40, 0x13 },
-	{ 0x10007a41, 0x77 },
-	{ 0x10007a42, 0x27 },
-	{ 0x10007a43, 0x00 },
-	{ 0x10007a44, 0x63 },
-	{ 0x10007a45, 0x18 },
-	{ 0x10007a46, 0x07 },
-	{ 0x10007a47, 0x00 },
-	{ 0x10007a48, 0x03 },
-	{ 0x10007a49, 0x47 },
-	{ 0x10007a4a, 0x36 },
-	{ 0x10007a4b, 0x54 },
-	{ 0x10007a4c, 0x13 },
-	{ 0x10007a4d, 0x77 },
-	{ 0x10007a4e, 0x17 },
-	{ 0x10007a4f, 0x00 },
-	{ 0x10007a50, 0xb3 },
-	{ 0x10007a51, 0x86 },
-	{ 0x10007a52, 0xe6 },
-	{ 0x10007a53, 0x40 },
-	{ 0x10007a54, 0x23 },
-	{ 0x10007a55, 0x85 },
-	{ 0x10007a56, 0xd1 },
-	{ 0x10007a57, 0x42 },
-	{ 0x10007a58, 0xb7 },
-	{ 0x10007a59, 0xd5 },
-	{ 0x10007a5a, 0x00 },
-	{ 0x10007a5b, 0x00 },
-	{ 0x10007a5c, 0x03 },
-	{ 0x10007a5d, 0xc6 },
-	{ 0x10007a5e, 0x05 },
-	{ 0x10007a5f, 0x92 },
-	{ 0x10007a60, 0x13 },
-	{ 0x10007a61, 0x07 },
-	{ 0x10007a62, 0x10 },
-	{ 0x10007a63, 0x00 },
-	{ 0x10007a64, 0x13 },
-	{ 0x10007a65, 0x76 },
-	{ 0x10007a66, 0x16 },
-	{ 0x10007a67, 0x00 },
-	{ 0x10007a68, 0x63 },
-	{ 0x10007a69, 0x1c },
-	{ 0x10007a6a, 0x06 },
-	{ 0x10007a6b, 0x00 },
-	{ 0x10007a6c, 0x03 },
-	{ 0x10007a6d, 0xc7 },
-	{ 0x10007a6e, 0x35 },
-	{ 0x10007a6f, 0x54 },
-	{ 0x10007a70, 0x13 },
-	{ 0x10007a71, 0x77 },
-	{ 0x10007a72, 0xf7 },
-	{ 0x10007a73, 0x0f },
-	{ 0x10007a74, 0x13 },
-	{ 0x10007a75, 0x57 },
-	{ 0x10007a76, 0x37 },
-	{ 0x10007a77, 0x00 },
-	{ 0x10007a78, 0x13 },
-	{ 0x10007a79, 0x47 },
-	{ 0x10007a7a, 0x17 },
-	{ 0x10007a7b, 0x00 },
-	{ 0x10007a7c, 0x13 },
-	{ 0x10007a7d, 0x77 },
-	{ 0x10007a7e, 0x17 },
-	{ 0x10007a7f, 0x00 },
-	{ 0x10007a80, 0xa3 },
-	{ 0x10007a81, 0x84 },
-	{ 0x10007a82, 0xe1 },
-	{ 0x10007a83, 0x42 },
-	{ 0x10007a84, 0xb7 },
-	{ 0x10007a85, 0xd5 },
-	{ 0x10007a86, 0x00 },
-	{ 0x10007a87, 0x00 },
-	{ 0x10007a88, 0x03 },
-	{ 0x10007a89, 0xc6 },
-	{ 0x10007a8a, 0x05 },
-	{ 0x10007a8b, 0x92 },
-	{ 0x10007a8c, 0x13 },
-	{ 0x10007a8d, 0x07 },
-	{ 0x10007a8e, 0x10 },
-	{ 0x10007a8f, 0x00 },
-	{ 0x10007a90, 0x13 },
-	{ 0x10007a91, 0x76 },
-	{ 0x10007a92, 0x26 },
-	{ 0x10007a93, 0x00 },
-	{ 0x10007a94, 0x63 },
-	{ 0x10007a95, 0x1c },
-	{ 0x10007a96, 0x06 },
-	{ 0x10007a97, 0x00 },
-	{ 0x10007a98, 0x03 },
-	{ 0x10007a99, 0xc7 },
-	{ 0x10007a9a, 0x35 },
-	{ 0x10007a9b, 0x54 },
-	{ 0x10007a9c, 0x13 },
-	{ 0x10007a9d, 0x77 },
-	{ 0x10007a9e, 0xf7 },
-	{ 0x10007a9f, 0x0f },
-	{ 0x10007aa0, 0x13 },
-	{ 0x10007aa1, 0x57 },
-	{ 0x10007aa2, 0x27 },
-	{ 0x10007aa3, 0x00 },
-	{ 0x10007aa4, 0x13 },
-	{ 0x10007aa5, 0x47 },
-	{ 0x10007aa6, 0x17 },
-	{ 0x10007aa7, 0x00 },
-	{ 0x10007aa8, 0x13 },
-	{ 0x10007aa9, 0x77 },
-	{ 0x10007aaa, 0x17 },
-	{ 0x10007aab, 0x00 },
-	{ 0x10007aac, 0x23 },
-	{ 0x10007aad, 0x84 },
-	{ 0x10007aae, 0xe1 },
-	{ 0x10007aaf, 0x42 },
-	{ 0x10007ab0, 0x63 },
-	{ 0x10007ab1, 0x84 },
-	{ 0x10007ab2, 0x07 },
-	{ 0x10007ab3, 0x00 },
-	{ 0x10007ab4, 0xe3 },
-	{ 0x10007ab5, 0x94 },
-	{ 0x10007ab6, 0x06 },
-	{ 0x10007ab7, 0xee },
-	{ 0x10007ab8, 0xef },
-	{ 0x10007ab9, 0x90 },
-	{ 0x10007aba, 0x0f },
-	{ 0x10007abb, 0x86 },
-	{ 0x10007abc, 0xef },
-	{ 0x10007abd, 0xd0 },
-	{ 0x10007abe, 0x0f },
-	{ 0x10007abf, 0x97 },
-	{ 0x10007ac0, 0x37 },
-	{ 0x10007ac1, 0x15 },
-	{ 0x10007ac2, 0x00 },
-	{ 0x10007ac3, 0x00 },
-	{ 0x10007ac4, 0x13 },
-	{ 0x10007ac5, 0x05 },
-	{ 0x10007ac6, 0x85 },
-	{ 0x10007ac7, 0xbb },
-	{ 0x10007ac8, 0x93 },
-	{ 0x10007ac9, 0x05 },
-	{ 0x10007aca, 0x00 },
-	{ 0x10007acb, 0x00 },
-	{ 0x10007acc, 0xef },
-	{ 0x10007acd, 0xd0 },
-	{ 0x10007ace, 0x9f },
-	{ 0x10007acf, 0xf5 },
-	{ 0x10007ad0, 0xb7 },
-	{ 0x10007ad1, 0xd7 },
-	{ 0x10007ad2, 0x00 },
-	{ 0x10007ad3, 0x00 },
-	{ 0x10007ad4, 0x83 },
-	{ 0x10007ad5, 0xc7 },
-	{ 0x10007ad6, 0x07 },
-	{ 0x10007ad7, 0x47 },
-	{ 0x10007ad8, 0x93 },
-	{ 0x10007ad9, 0xf7 },
-	{ 0x10007ada, 0x47 },
-	{ 0x10007adb, 0x00 },
-	{ 0x10007adc, 0xe3 },
-	{ 0x10007add, 0x80 },
-	{ 0x10007ade, 0x07 },
-	{ 0x10007adf, 0xec },
-	{ 0x10007ae0, 0xef },
-	{ 0x10007ae1, 0x80 },
-	{ 0x10007ae2, 0xdf },
-	{ 0x10007ae3, 0xf4 },
-	{ 0x10007ae4, 0x23 },
-	{ 0x10007ae5, 0x89 },
-	{ 0x10007ae6, 0xa1 },
-	{ 0x10007ae7, 0x40 },
-	{ 0x10007ae8, 0x6f },
-	{ 0x10007ae9, 0xf0 },
-	{ 0x10007aea, 0x5f },
-	{ 0x10007aeb, 0xeb },
-	{ 0x10007aec, 0x00 },
-	{ 0x10007aed, 0x00 },
-	{ 0x10007aee, 0x00 },
-	{ 0x10007aef, 0x00 },
 	{ 0x3fc2bf83, 0x00 },
 	{ 0x3fc2bf82, 0x00 },
 	{ 0x3fc2bf81, 0x00 },
@@ -2112,1330 +228,6 @@ static const struct reg_sequence rt1320_vc_patch_code_write[] = {
 	{ SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT_PDE23, RT1320_SDCA_CTL_REQ_POWER_STATE, 0), 0x03 },
 };
 
-/*
- * The 'patch code' is written to the patch code area.
- * The patch code area is used for SDCA register expansion flexibility.
- */
-static const struct reg_sequence rt1320_patch_code_write[] = {
-	{ 0x10007000, 0x37 },
-	{ 0x10007001, 0x77 },
-	{ 0x10007002, 0x00 },
-	{ 0x10007003, 0x10 },
-	{ 0x10007004, 0xb7 },
-	{ 0x10007005, 0xe7 },
-	{ 0x10007006, 0x00 },
-	{ 0x10007007, 0x10 },
-	{ 0x10007008, 0x13 },
-	{ 0x10007009, 0x07 },
-	{ 0x1000700a, 0x07 },
-	{ 0x1000700b, 0x40 },
-	{ 0x1000700c, 0x23 },
-	{ 0x1000700d, 0xae },
-	{ 0x1000700e, 0xe7 },
-	{ 0x1000700f, 0xda },
-	{ 0x10007010, 0x37 },
-	{ 0x10007011, 0x77 },
-	{ 0x10007012, 0x00 },
-	{ 0x10007013, 0x10 },
-	{ 0x10007014, 0x13 },
-	{ 0x10007015, 0x07 },
-	{ 0x10007016, 0x47 },
-	{ 0x10007017, 0x61 },
-	{ 0x10007018, 0x23 },
-	{ 0x10007019, 0xa4 },
-	{ 0x1000701a, 0xe7 },
-	{ 0x1000701b, 0xde },
-	{ 0x1000701c, 0x37 },
-	{ 0x1000701d, 0x77 },
-	{ 0x1000701e, 0x00 },
-	{ 0x1000701f, 0x10 },
-	{ 0x10007020, 0x13 },
-	{ 0x10007021, 0x07 },
-	{ 0x10007022, 0x07 },
-	{ 0x10007023, 0x52 },
-	{ 0x10007024, 0x23 },
-	{ 0x10007025, 0xae },
-	{ 0x10007026, 0xe7 },
-	{ 0x10007027, 0xde },
-	{ 0x10007028, 0x37 },
-	{ 0x10007029, 0x77 },
-	{ 0x1000702a, 0x00 },
-	{ 0x1000702b, 0x10 },
-	{ 0x1000702c, 0x13 },
-	{ 0x1000702d, 0x07 },
-	{ 0x1000702e, 0x47 },
-	{ 0x1000702f, 0x54 },
-	{ 0x10007030, 0x23 },
-	{ 0x10007031, 0xaa },
-	{ 0x10007032, 0xe7 },
-	{ 0x10007033, 0xe4 },
-	{ 0x10007034, 0x37 },
-	{ 0x10007035, 0x87 },
-	{ 0x10007036, 0x00 },
-	{ 0x10007037, 0x10 },
-	{ 0x10007038, 0x13 },
-	{ 0x10007039, 0x07 },
-	{ 0x1000703a, 0x47 },
-	{ 0x1000703b, 0x81 },
-	{ 0x1000703c, 0x23 },
-	{ 0x1000703d, 0xa2 },
-	{ 0x1000703e, 0xe7 },
-	{ 0x1000703f, 0xe8 },
-	{ 0x10007040, 0x23 },
-	{ 0x10007041, 0xa4 },
-	{ 0x10007042, 0xe7 },
-	{ 0x10007043, 0xe8 },
-	{ 0x10007044, 0x37 },
-	{ 0x10007045, 0x77 },
-	{ 0x10007046, 0x00 },
-	{ 0x10007047, 0x10 },
-	{ 0x10007048, 0x13 },
-	{ 0x10007049, 0x07 },
-	{ 0x1000704a, 0x07 },
-	{ 0x1000704b, 0x59 },
-	{ 0x1000704c, 0x23 },
-	{ 0x1000704d, 0xa8 },
-	{ 0x1000704e, 0xe7 },
-	{ 0x1000704f, 0xea },
-	{ 0x10007050, 0x37 },
-	{ 0x10007051, 0x77 },
-	{ 0x10007052, 0x00 },
-	{ 0x10007053, 0x10 },
-	{ 0x10007054, 0x13 },
-	{ 0x10007055, 0x07 },
-	{ 0x10007056, 0x07 },
-	{ 0x10007057, 0x78 },
-	{ 0x10007058, 0x23 },
-	{ 0x10007059, 0xa6 },
-	{ 0x1000705a, 0xe7 },
-	{ 0x1000705b, 0xec },
-	{ 0x1000705c, 0x67 },
-	{ 0x1000705d, 0x80 },
-	{ 0x1000705e, 0x00 },
-	{ 0x1000705f, 0x00 },
-	{ 0x10007400, 0x37 },
-	{ 0x10007401, 0xd7 },
-	{ 0x10007402, 0x00 },
-	{ 0x10007403, 0x00 },
-	{ 0x10007404, 0x83 },
-	{ 0x10007405, 0x27 },
-	{ 0x10007406, 0x47 },
-	{ 0x10007407, 0x56 },
-	{ 0x10007408, 0xb7 },
-	{ 0x10007409, 0x06 },
-	{ 0x1000740a, 0x00 },
-	{ 0x1000740b, 0x02 },
-	{ 0x1000740c, 0xb3 },
-	{ 0x1000740d, 0xf7 },
-	{ 0x1000740e, 0xd7 },
-	{ 0x1000740f, 0x00 },
-	{ 0x10007410, 0x63 },
-	{ 0x10007411, 0x8a },
-	{ 0x10007412, 0x07 },
-	{ 0x10007413, 0x00 },
-	{ 0x10007414, 0x93 },
-	{ 0x10007415, 0x06 },
-	{ 0x10007416, 0x10 },
-	{ 0x10007417, 0x00 },
-	{ 0x10007418, 0x23 },
-	{ 0x10007419, 0x83 },
-	{ 0x1000741a, 0xd1 },
-	{ 0x1000741b, 0x44 },
-	{ 0x1000741c, 0x93 },
-	{ 0x1000741d, 0x07 },
-	{ 0x1000741e, 0xf0 },
-	{ 0x1000741f, 0xff },
-	{ 0x10007420, 0x23 },
-	{ 0x10007421, 0x22 },
-	{ 0x10007422, 0xf7 },
-	{ 0x10007423, 0x56 },
-	{ 0x10007424, 0x37 },
-	{ 0x10007425, 0xd7 },
-	{ 0x10007426, 0x00 },
-	{ 0x10007427, 0x00 },
-	{ 0x10007428, 0x83 },
-	{ 0x10007429, 0x27 },
-	{ 0x1000742a, 0x47 },
-	{ 0x1000742b, 0x58 },
-	{ 0x1000742c, 0x93 },
-	{ 0x1000742d, 0xf7 },
-	{ 0x1000742e, 0x17 },
-	{ 0x1000742f, 0x00 },
-	{ 0x10007430, 0x63 },
-	{ 0x10007431, 0x86 },
-	{ 0x10007432, 0x07 },
-	{ 0x10007433, 0x00 },
-	{ 0x10007434, 0x93 },
-	{ 0x10007435, 0x07 },
-	{ 0x10007436, 0x10 },
-	{ 0x10007437, 0x00 },
-	{ 0x10007438, 0x23 },
-	{ 0x10007439, 0x22 },
-	{ 0x1000743a, 0xf7 },
-	{ 0x1000743b, 0x58 },
-	{ 0x1000743c, 0xb7 },
-	{ 0x1000743d, 0xd7 },
-	{ 0x1000743e, 0x00 },
-	{ 0x1000743f, 0x00 },
-	{ 0x10007440, 0x03 },
-	{ 0x10007441, 0xa7 },
-	{ 0x10007442, 0x47 },
-	{ 0x10007443, 0x58 },
-	{ 0x10007444, 0xb7 },
-	{ 0x10007445, 0x07 },
-	{ 0x10007446, 0x00 },
-	{ 0x10007447, 0x04 },
-	{ 0x10007448, 0x33 },
-	{ 0x10007449, 0x77 },
-	{ 0x1000744a, 0xf7 },
-	{ 0x1000744b, 0x00 },
-	{ 0x1000744c, 0x93 },
-	{ 0x1000744d, 0x07 },
-	{ 0x1000744e, 0x00 },
-	{ 0x1000744f, 0x00 },
-	{ 0x10007450, 0x63 },
-	{ 0x10007451, 0x0e },
-	{ 0x10007452, 0x07 },
-	{ 0x10007453, 0x04 },
-	{ 0x10007454, 0x37 },
-	{ 0x10007455, 0x07 },
-	{ 0x10007456, 0x00 },
-	{ 0x10007457, 0x11 },
-	{ 0x10007458, 0x03 },
-	{ 0x10007459, 0x47 },
-	{ 0x1000745a, 0x87 },
-	{ 0x1000745b, 0x0e },
-	{ 0x1000745c, 0x93 },
-	{ 0x1000745d, 0x06 },
-	{ 0x1000745e, 0x40 },
-	{ 0x1000745f, 0x00 },
-	{ 0x10007460, 0x13 },
-	{ 0x10007461, 0x77 },
-	{ 0x10007462, 0xf7 },
-	{ 0x10007463, 0x0f },
-	{ 0x10007464, 0x63 },
-	{ 0x10007465, 0x02 },
-	{ 0x10007466, 0xd7 },
-	{ 0x10007467, 0x0a },
-	{ 0x10007468, 0x93 },
-	{ 0x10007469, 0x06 },
-	{ 0x1000746a, 0x70 },
-	{ 0x1000746b, 0x00 },
-	{ 0x1000746c, 0x63 },
-	{ 0x1000746d, 0x10 },
-	{ 0x1000746e, 0xd7 },
-	{ 0x1000746f, 0x04 },
-	{ 0x10007470, 0x93 },
-	{ 0x10007471, 0x07 },
-	{ 0x10007472, 0x60 },
-	{ 0x10007473, 0x06 },
-	{ 0x10007474, 0x37 },
-	{ 0x10007475, 0xd7 },
-	{ 0x10007476, 0x00 },
-	{ 0x10007477, 0x00 },
-	{ 0x10007478, 0x83 },
-	{ 0x10007479, 0x46 },
-	{ 0x1000747a, 0x77 },
-	{ 0x1000747b, 0xa6 },
-	{ 0x1000747c, 0x93 },
-	{ 0x1000747d, 0xe6 },
-	{ 0x1000747e, 0x06 },
-	{ 0x1000747f, 0xf8 },
-	{ 0x10007480, 0x93 },
-	{ 0x10007481, 0xf6 },
-	{ 0x10007482, 0xf6 },
-	{ 0x10007483, 0x0f },
-	{ 0x10007484, 0xa3 },
-	{ 0x10007485, 0x03 },
-	{ 0x10007486, 0xd7 },
-	{ 0x10007487, 0xa6 },
-	{ 0x10007488, 0x83 },
-	{ 0x10007489, 0x46 },
-	{ 0x1000748a, 0x77 },
-	{ 0x1000748b, 0xa8 },
-	{ 0x1000748c, 0x93 },
-	{ 0x1000748d, 0xe6 },
-	{ 0x1000748e, 0x06 },
-	{ 0x1000748f, 0xf8 },
-	{ 0x10007490, 0x93 },
-	{ 0x10007491, 0xf6 },
-	{ 0x10007492, 0xf6 },
-	{ 0x10007493, 0x0f },
-	{ 0x10007494, 0xa3 },
-	{ 0x10007495, 0x03 },
-	{ 0x10007496, 0xd7 },
-	{ 0x10007497, 0xa8 },
-	{ 0x10007498, 0xb7 },
-	{ 0x10007499, 0xc6 },
-	{ 0x1000749a, 0x00 },
-	{ 0x1000749b, 0x00 },
-	{ 0x1000749c, 0x23 },
-	{ 0x1000749d, 0x84 },
-	{ 0x1000749e, 0xf6 },
-	{ 0x1000749f, 0x06 },
-	{ 0x100074a0, 0xa3 },
-	{ 0x100074a1, 0x84 },
-	{ 0x100074a2, 0xf6 },
-	{ 0x100074a3, 0x06 },
-	{ 0x100074a4, 0xb7 },
-	{ 0x100074a5, 0x06 },
-	{ 0x100074a6, 0x00 },
-	{ 0x100074a7, 0x04 },
-	{ 0x100074a8, 0x23 },
-	{ 0x100074a9, 0x22 },
-	{ 0x100074aa, 0xd7 },
-	{ 0x100074ab, 0x58 },
-	{ 0x100074ac, 0x37 },
-	{ 0x100074ad, 0xd7 },
-	{ 0x100074ae, 0x00 },
-	{ 0x100074af, 0x00 },
-	{ 0x100074b0, 0x03 },
-	{ 0x100074b1, 0x27 },
-	{ 0x100074b2, 0x47 },
-	{ 0x100074b3, 0x58 },
-	{ 0x100074b4, 0xb7 },
-	{ 0x100074b5, 0x06 },
-	{ 0x100074b6, 0x00 },
-	{ 0x100074b7, 0x08 },
-	{ 0x100074b8, 0x33 },
-	{ 0x100074b9, 0x77 },
-	{ 0x100074ba, 0xd7 },
-	{ 0x100074bb, 0x00 },
-	{ 0x100074bc, 0x63 },
-	{ 0x100074bd, 0x04 },
-	{ 0x100074be, 0x07 },
-	{ 0x100074bf, 0x04 },
-	{ 0x100074c0, 0x37 },
-	{ 0x100074c1, 0x07 },
-	{ 0x100074c2, 0x00 },
-	{ 0x100074c3, 0x11 },
-	{ 0x100074c4, 0x03 },
-	{ 0x100074c5, 0x47 },
-	{ 0x100074c6, 0xc7 },
-	{ 0x100074c7, 0x0e },
-	{ 0x100074c8, 0x93 },
-	{ 0x100074c9, 0x06 },
-	{ 0x100074ca, 0x40 },
-	{ 0x100074cb, 0x00 },
-	{ 0x100074cc, 0x13 },
-	{ 0x100074cd, 0x77 },
-	{ 0x100074ce, 0xf7 },
-	{ 0x100074cf, 0x0f },
-	{ 0x100074d0, 0x63 },
-	{ 0x100074d1, 0x00 },
-	{ 0x100074d2, 0xd7 },
-	{ 0x100074d3, 0x04 },
-	{ 0x100074d4, 0x93 },
-	{ 0x100074d5, 0x06 },
-	{ 0x100074d6, 0x70 },
-	{ 0x100074d7, 0x00 },
-	{ 0x100074d8, 0x63 },
-	{ 0x100074d9, 0x00 },
-	{ 0x100074da, 0xd7 },
-	{ 0x100074db, 0x04 },
-	{ 0x100074dc, 0x63 },
-	{ 0x100074dd, 0x84 },
-	{ 0x100074de, 0x07 },
-	{ 0x100074df, 0x02 },
-	{ 0x100074e0, 0xb7 },
-	{ 0x100074e1, 0xd6 },
-	{ 0x100074e2, 0x00 },
-	{ 0x100074e3, 0x00 },
-	{ 0x100074e4, 0x03 },
-	{ 0x100074e5, 0xc7 },
-	{ 0x100074e6, 0x56 },
-	{ 0x100074e7, 0xa4 },
-	{ 0x100074e8, 0x13 },
-	{ 0x100074e9, 0x67 },
-	{ 0x100074ea, 0x07 },
-	{ 0x100074eb, 0xf8 },
-	{ 0x100074ec, 0x13 },
-	{ 0x100074ed, 0x77 },
-	{ 0x100074ee, 0xf7 },
-	{ 0x100074ef, 0x0f },
-	{ 0x100074f0, 0xa3 },
-	{ 0x100074f1, 0x82 },
-	{ 0x100074f2, 0xe6 },
-	{ 0x100074f3, 0xa4 },
-	{ 0x100074f4, 0x37 },
-	{ 0x100074f5, 0xc7 },
-	{ 0x100074f6, 0x00 },
-	{ 0x100074f7, 0x00 },
-	{ 0x100074f8, 0x23 },
-	{ 0x100074f9, 0x02 },
-	{ 0x100074fa, 0xf7 },
-	{ 0x100074fb, 0x06 },
-	{ 0x100074fc, 0xb7 },
-	{ 0x100074fd, 0x07 },
-	{ 0x100074fe, 0x00 },
-	{ 0x100074ff, 0x08 },
-	{ 0x10007500, 0x23 },
-	{ 0x10007501, 0xa2 },
-	{ 0x10007502, 0xf6 },
-	{ 0x10007503, 0x58 },
-	{ 0x10007504, 0x67 },
-	{ 0x10007505, 0x80 },
-	{ 0x10007506, 0x00 },
-	{ 0x10007507, 0x00 },
-	{ 0x10007508, 0x93 },
-	{ 0x10007509, 0x07 },
-	{ 0x1000750a, 0x80 },
-	{ 0x1000750b, 0x08 },
-	{ 0x1000750c, 0x6f },
-	{ 0x1000750d, 0xf0 },
-	{ 0x1000750e, 0x9f },
-	{ 0x1000750f, 0xf6 },
-	{ 0x10007510, 0x93 },
-	{ 0x10007511, 0x07 },
-	{ 0x10007512, 0x80 },
-	{ 0x10007513, 0x08 },
-	{ 0x10007514, 0x6f },
-	{ 0x10007515, 0xf0 },
-	{ 0x10007516, 0xdf },
-	{ 0x10007517, 0xfc },
-	{ 0x10007518, 0x93 },
-	{ 0x10007519, 0x07 },
-	{ 0x1000751a, 0x60 },
-	{ 0x1000751b, 0x06 },
-	{ 0x1000751c, 0x6f },
-	{ 0x1000751d, 0xf0 },
-	{ 0x1000751e, 0x5f },
-	{ 0x1000751f, 0xfc },
-	{ 0x10007520, 0x37 },
-	{ 0x10007521, 0xd7 },
-	{ 0x10007522, 0x00 },
-	{ 0x10007523, 0x00 },
-	{ 0x10007524, 0x83 },
-	{ 0x10007525, 0x27 },
-	{ 0x10007526, 0x07 },
-	{ 0x10007527, 0x53 },
-	{ 0x10007528, 0xb7 },
-	{ 0x10007529, 0x06 },
-	{ 0x1000752a, 0x02 },
-	{ 0x1000752b, 0x00 },
-	{ 0x1000752c, 0xb3 },
-	{ 0x1000752d, 0xf7 },
-	{ 0x1000752e, 0xd7 },
-	{ 0x1000752f, 0x00 },
-	{ 0x10007530, 0x63 },
-	{ 0x10007531, 0x88 },
-	{ 0x10007532, 0x07 },
-	{ 0x10007533, 0x00 },
-	{ 0x10007534, 0x13 },
-	{ 0x10007535, 0x06 },
-	{ 0x10007536, 0xa0 },
-	{ 0x10007537, 0x05 },
-	{ 0x10007538, 0x23 },
-	{ 0x10007539, 0xa8 },
-	{ 0x1000753a, 0xc1 },
-	{ 0x1000753b, 0x56 },
-	{ 0x1000753c, 0x23 },
-	{ 0x1000753d, 0x28 },
-	{ 0x1000753e, 0xd7 },
-	{ 0x1000753f, 0x52 },
-	{ 0x10007540, 0x67 },
-	{ 0x10007541, 0x80 },
-	{ 0x10007542, 0x00 },
-	{ 0x10007543, 0x00 },
-	{ 0x10007544, 0x37 },
-	{ 0x10007545, 0xd7 },
-	{ 0x10007546, 0x00 },
-	{ 0x10007547, 0x10 },
-	{ 0x10007548, 0x83 },
-	{ 0x10007549, 0x47 },
-	{ 0x1000754a, 0x07 },
-	{ 0x1000754b, 0xd9 },
-	{ 0x1000754c, 0x93 },
-	{ 0x1000754d, 0x06 },
-	{ 0x1000754e, 0x20 },
-	{ 0x1000754f, 0x00 },
-	{ 0x10007550, 0x93 },
-	{ 0x10007551, 0xf7 },
-	{ 0x10007552, 0xf7 },
-	{ 0x10007553, 0x0f },
-	{ 0x10007554, 0x63 },
-	{ 0x10007555, 0x9c },
-	{ 0x10007556, 0xd7 },
-	{ 0x10007557, 0x02 },
-	{ 0x10007558, 0xb7 },
-	{ 0x10007559, 0xc6 },
-	{ 0x1000755a, 0x00 },
-	{ 0x1000755b, 0x00 },
-	{ 0x1000755c, 0x83 },
-	{ 0x1000755d, 0xc7 },
-	{ 0x1000755e, 0x26 },
-	{ 0x1000755f, 0x04 },
-	{ 0x10007560, 0x93 },
-	{ 0x10007561, 0xf7 },
-	{ 0x10007562, 0xf7 },
-	{ 0x10007563, 0x07 },
-	{ 0x10007564, 0x23 },
-	{ 0x10007565, 0x81 },
-	{ 0x10007566, 0xf6 },
-	{ 0x10007567, 0x04 },
-	{ 0x10007568, 0xb7 },
-	{ 0x10007569, 0xd6 },
-	{ 0x1000756a, 0x00 },
-	{ 0x1000756b, 0x00 },
-	{ 0x1000756c, 0x83 },
-	{ 0x1000756d, 0xc7 },
-	{ 0x1000756e, 0xa6 },
-	{ 0x1000756f, 0xe1 },
-	{ 0x10007570, 0x93 },
-	{ 0x10007571, 0xf7 },
-	{ 0x10007572, 0xf7 },
-	{ 0x10007573, 0x07 },
-	{ 0x10007574, 0x23 },
-	{ 0x10007575, 0x8d },
-	{ 0x10007576, 0xf6 },
-	{ 0x10007577, 0xe0 },
-	{ 0x10007578, 0x23 },
-	{ 0x10007579, 0x08 },
-	{ 0x1000757a, 0x07 },
-	{ 0x1000757b, 0xd8 },
-	{ 0x1000757c, 0x83 },
-	{ 0x1000757d, 0x47 },
-	{ 0x1000757e, 0x47 },
-	{ 0x1000757f, 0xd9 },
-	{ 0x10007580, 0x93 },
-	{ 0x10007581, 0x87 },
-	{ 0x10007582, 0x17 },
-	{ 0x10007583, 0x00 },
-	{ 0x10007584, 0x93 },
-	{ 0x10007585, 0xf7 },
-	{ 0x10007586, 0xf7 },
-	{ 0x10007587, 0x0f },
-	{ 0x10007588, 0x23 },
-	{ 0x10007589, 0x0a },
-	{ 0x1000758a, 0xf7 },
-	{ 0x1000758b, 0xd8 },
-	{ 0x1000758c, 0x67 },
-	{ 0x1000758d, 0x80 },
-	{ 0x1000758e, 0x00 },
-	{ 0x1000758f, 0x00 },
-	{ 0x10007590, 0xb7 },
-	{ 0x10007591, 0xd7 },
-	{ 0x10007592, 0x00 },
-	{ 0x10007593, 0x00 },
-	{ 0x10007594, 0x83 },
-	{ 0x10007595, 0xc7 },
-	{ 0x10007596, 0x07 },
-	{ 0x10007597, 0x47 },
-	{ 0x10007598, 0x93 },
-	{ 0x10007599, 0xf7 },
-	{ 0x1000759a, 0x07 },
-	{ 0x1000759b, 0x01 },
-	{ 0x1000759c, 0x63 },
-	{ 0x1000759d, 0x8a },
-	{ 0x1000759e, 0x07 },
-	{ 0x1000759f, 0x06 },
-	{ 0x100075a0, 0x63 },
-	{ 0x100075a1, 0x02 },
-	{ 0x100075a2, 0x05 },
-	{ 0x100075a3, 0x06 },
-	{ 0x100075a4, 0x37 },
-	{ 0x100075a5, 0xc7 },
-	{ 0x100075a6, 0x00 },
-	{ 0x100075a7, 0x00 },
-	{ 0x100075a8, 0x83 },
-	{ 0x100075a9, 0x27 },
-	{ 0x100075aa, 0xc7 },
-	{ 0x100075ab, 0x5f },
-	{ 0x100075ac, 0x23 },
-	{ 0x100075ad, 0xae },
-	{ 0x100075ae, 0xf1 },
-	{ 0x100075af, 0x40 },
-	{ 0x100075b0, 0xb7 },
-	{ 0x100075b1, 0x06 },
-	{ 0x100075b2, 0x00 },
-	{ 0x100075b3, 0x10 },
-	{ 0x100075b4, 0xb3 },
-	{ 0x100075b5, 0xf7 },
-	{ 0x100075b6, 0xd7 },
-	{ 0x100075b7, 0x00 },
-	{ 0x100075b8, 0x63 },
-	{ 0x100075b9, 0x8c },
-	{ 0x100075ba, 0x07 },
-	{ 0x100075bb, 0x04 },
-	{ 0x100075bc, 0x83 },
-	{ 0x100075bd, 0x47 },
-	{ 0x100075be, 0x07 },
-	{ 0x100075bf, 0x56 },
-	{ 0x100075c0, 0x93 },
-	{ 0x100075c1, 0xf7 },
-	{ 0x100075c2, 0x87 },
-	{ 0x100075c3, 0x01 },
-	{ 0x100075c4, 0x63 },
-	{ 0x100075c5, 0x86 },
-	{ 0x100075c6, 0x07 },
-	{ 0x100075c7, 0x04 },
-	{ 0x100075c8, 0x83 },
-	{ 0x100075c9, 0x47 },
-	{ 0x100075ca, 0x17 },
-	{ 0x100075cb, 0x08 },
-	{ 0x100075cc, 0x93 },
-	{ 0x100075cd, 0xf7 },
-	{ 0x100075ce, 0x47 },
-	{ 0x100075cf, 0x00 },
-	{ 0x100075d0, 0x63 },
-	{ 0x100075d1, 0x80 },
-	{ 0x100075d2, 0x07 },
-	{ 0x100075d3, 0x04 },
-	{ 0x100075d4, 0xb7 },
-	{ 0x100075d5, 0xc7 },
-	{ 0x100075d6, 0xc2 },
-	{ 0x100075d7, 0x3f },
-	{ 0x100075d8, 0x93 },
-	{ 0x100075d9, 0x87 },
-	{ 0x100075da, 0x07 },
-	{ 0x100075db, 0xfc },
-	{ 0x100075dc, 0x83 },
-	{ 0x100075dd, 0xa7 },
-	{ 0x100075de, 0x47 },
-	{ 0x100075df, 0x00 },
-	{ 0x100075e0, 0x93 },
-	{ 0x100075e1, 0xd7 },
-	{ 0x100075e2, 0x17 },
-	{ 0x100075e3, 0x00 },
-	{ 0x100075e4, 0x93 },
-	{ 0x100075e5, 0xf7 },
-	{ 0x100075e6, 0x17 },
-	{ 0x100075e7, 0x00 },
-	{ 0x100075e8, 0x63 },
-	{ 0x100075e9, 0x84 },
-	{ 0x100075ea, 0x07 },
-	{ 0x100075eb, 0x02 },
-	{ 0x100075ec, 0x23 },
-	{ 0x100075ed, 0x8a },
-	{ 0x100075ee, 0xf1 },
-	{ 0x100075ef, 0x40 },
-	{ 0x100075f0, 0xb7 },
-	{ 0x100075f1, 0x07 },
-	{ 0x100075f2, 0x00 },
-	{ 0x100075f3, 0xc0 },
-	{ 0x100075f4, 0x37 },
-	{ 0x100075f5, 0xf7 },
-	{ 0x100075f6, 0x00 },
-	{ 0x100075f7, 0x00 },
-	{ 0x100075f8, 0x93 },
-	{ 0x100075f9, 0x87 },
-	{ 0x100075fa, 0xf7 },
-	{ 0x100075fb, 0xff },
-	{ 0x100075fc, 0x23 },
-	{ 0x100075fd, 0x2c },
-	{ 0x100075fe, 0xf7 },
-	{ 0x100075ff, 0x06 },
-	{ 0x10007600, 0x67 },
-	{ 0x10007601, 0x80 },
-	{ 0x10007602, 0x00 },
-	{ 0x10007603, 0x00 },
-	{ 0x10007604, 0x23 },
-	{ 0x10007605, 0x8a },
-	{ 0x10007606, 0x01 },
-	{ 0x10007607, 0x40 },
-	{ 0x10007608, 0xb7 },
-	{ 0x10007609, 0xf7 },
-	{ 0x1000760a, 0x00 },
-	{ 0x1000760b, 0x00 },
-	{ 0x1000760c, 0x23 },
-	{ 0x1000760d, 0xac },
-	{ 0x1000760e, 0x07 },
-	{ 0x1000760f, 0x06 },
-	{ 0x10007610, 0x67 },
-	{ 0x10007611, 0x80 },
-	{ 0x10007612, 0x00 },
-	{ 0x10007613, 0x00 },
-	{ 0x10007614, 0x13 },
-	{ 0x10007615, 0x01 },
-	{ 0x10007616, 0x01 },
-	{ 0x10007617, 0xff },
-	{ 0x10007618, 0x23 },
-	{ 0x10007619, 0x26 },
-	{ 0x1000761a, 0x11 },
-	{ 0x1000761b, 0x00 },
-	{ 0x1000761c, 0x23 },
-	{ 0x1000761d, 0x24 },
-	{ 0x1000761e, 0x81 },
-	{ 0x1000761f, 0x00 },
-	{ 0x10007620, 0x37 },
-	{ 0x10007621, 0xc7 },
-	{ 0x10007622, 0x00 },
-	{ 0x10007623, 0x00 },
-	{ 0x10007624, 0x83 },
-	{ 0x10007625, 0x47 },
-	{ 0x10007626, 0x07 },
-	{ 0x10007627, 0x56 },
-	{ 0x10007628, 0x93 },
-	{ 0x10007629, 0xf7 },
-	{ 0x1000762a, 0x17 },
-	{ 0x1000762b, 0x00 },
-	{ 0x1000762c, 0x63 },
-	{ 0x1000762d, 0x98 },
-	{ 0x1000762e, 0x07 },
-	{ 0x1000762f, 0x00 },
-	{ 0x10007630, 0x83 },
-	{ 0x10007631, 0x47 },
-	{ 0x10007632, 0x07 },
-	{ 0x10007633, 0x56 },
-	{ 0x10007634, 0x93 },
-	{ 0x10007635, 0xf7 },
-	{ 0x10007636, 0x27 },
-	{ 0x10007637, 0x00 },
-	{ 0x10007638, 0x63 },
-	{ 0x10007639, 0x82 },
-	{ 0x1000763a, 0x07 },
-	{ 0x1000763b, 0x08 },
-	{ 0x1000763c, 0x37 },
-	{ 0x1000763d, 0xd4 },
-	{ 0x1000763e, 0x00 },
-	{ 0x1000763f, 0x00 },
-	{ 0x10007640, 0x83 },
-	{ 0x10007641, 0x47 },
-	{ 0x10007642, 0x14 },
-	{ 0x10007643, 0x47 },
-	{ 0x10007644, 0x93 },
-	{ 0x10007645, 0xf7 },
-	{ 0x10007646, 0x27 },
-	{ 0x10007647, 0x00 },
-	{ 0x10007648, 0x63 },
-	{ 0x10007649, 0x8a },
-	{ 0x1000764a, 0x07 },
-	{ 0x1000764b, 0x06 },
-	{ 0x1000764c, 0x93 },
-	{ 0x1000764d, 0x05 },
-	{ 0x1000764e, 0x10 },
-	{ 0x1000764f, 0x00 },
-	{ 0x10007650, 0x13 },
-	{ 0x10007651, 0x05 },
-	{ 0x10007652, 0x20 },
-	{ 0x10007653, 0x10 },
-	{ 0x10007654, 0xef },
-	{ 0x10007655, 0xa0 },
-	{ 0x10007656, 0x8f },
-	{ 0x10007657, 0x9a },
-	{ 0x10007658, 0x37 },
-	{ 0x10007659, 0x05 },
-	{ 0x1000765a, 0x01 },
-	{ 0x1000765b, 0x00 },
-	{ 0x1000765c, 0x93 },
-	{ 0x1000765d, 0x05 },
-	{ 0x1000765e, 0x00 },
-	{ 0x1000765f, 0x01 },
-	{ 0x10007660, 0x13 },
-	{ 0x10007661, 0x05 },
-	{ 0x10007662, 0xb5 },
-	{ 0x10007663, 0xa0 },
-	{ 0x10007664, 0xef },
-	{ 0x10007665, 0xa0 },
-	{ 0x10007666, 0x8f },
-	{ 0x10007667, 0x99 },
-	{ 0x10007668, 0x83 },
-	{ 0x10007669, 0x47 },
-	{ 0x1000766a, 0x24 },
-	{ 0x1000766b, 0xe0 },
-	{ 0x1000766c, 0x13 },
-	{ 0x1000766d, 0x05 },
-	{ 0x1000766e, 0x80 },
-	{ 0x1000766f, 0x3e },
-	{ 0x10007670, 0x93 },
-	{ 0x10007671, 0x05 },
-	{ 0x10007672, 0x00 },
-	{ 0x10007673, 0x00 },
-	{ 0x10007674, 0x93 },
-	{ 0x10007675, 0xe7 },
-	{ 0x10007676, 0x07 },
-	{ 0x10007677, 0xf8 },
-	{ 0x10007678, 0x93 },
-	{ 0x10007679, 0xf7 },
-	{ 0x1000767a, 0xf7 },
-	{ 0x1000767b, 0x0f },
-	{ 0x1000767c, 0x23 },
-	{ 0x1000767d, 0x01 },
-	{ 0x1000767e, 0xf4 },
-	{ 0x1000767f, 0xe0 },
-	{ 0x10007680, 0x83 },
-	{ 0x10007681, 0x47 },
-	{ 0x10007682, 0x24 },
-	{ 0x10007683, 0xe0 },
-	{ 0x10007684, 0x93 },
-	{ 0x10007685, 0xf7 },
-	{ 0x10007686, 0xf7 },
-	{ 0x10007687, 0x0f },
-	{ 0x10007688, 0x93 },
-	{ 0x10007689, 0xe7 },
-	{ 0x1000768a, 0x07 },
-	{ 0x1000768b, 0x04 },
-	{ 0x1000768c, 0x23 },
-	{ 0x1000768d, 0x01 },
-	{ 0x1000768e, 0xf4 },
-	{ 0x1000768f, 0xe0 },
-	{ 0x10007690, 0xef },
-	{ 0x10007691, 0xe0 },
-	{ 0x10007692, 0x8f },
-	{ 0x10007693, 0xb9 },
-	{ 0x10007694, 0x83 },
-	{ 0x10007695, 0x47 },
-	{ 0x10007696, 0x34 },
-	{ 0x10007697, 0xe0 },
-	{ 0x10007698, 0x93 },
-	{ 0x10007699, 0xf7 },
-	{ 0x1000769a, 0x07 },
-	{ 0x1000769b, 0x02 },
-	{ 0x1000769c, 0xe3 },
-	{ 0x1000769d, 0x9c },
-	{ 0x1000769e, 0x07 },
-	{ 0x1000769f, 0xfe },
-	{ 0x100076a0, 0x37 },
-	{ 0x100076a1, 0x05 },
-	{ 0x100076a2, 0x01 },
-	{ 0x100076a3, 0x00 },
-	{ 0x100076a4, 0x93 },
-	{ 0x100076a5, 0x05 },
-	{ 0x100076a6, 0x00 },
-	{ 0x100076a7, 0x00 },
-	{ 0x100076a8, 0x13 },
-	{ 0x100076a9, 0x05 },
-	{ 0x100076aa, 0xb5 },
-	{ 0x100076ab, 0xa0 },
-	{ 0x100076ac, 0xef },
-	{ 0x100076ad, 0xa0 },
-	{ 0x100076ae, 0x0f },
-	{ 0x100076af, 0x95 },
-	{ 0x100076b0, 0x83 },
-	{ 0x100076b1, 0x47 },
-	{ 0x100076b2, 0x14 },
-	{ 0x100076b3, 0x47 },
-	{ 0x100076b4, 0x93 },
-	{ 0x100076b5, 0xf7 },
-	{ 0x100076b6, 0xd7 },
-	{ 0x100076b7, 0x0f },
-	{ 0x100076b8, 0xa3 },
-	{ 0x100076b9, 0x08 },
-	{ 0x100076ba, 0xf4 },
-	{ 0x100076bb, 0x46 },
-	{ 0x100076bc, 0x03 },
-	{ 0x100076bd, 0xa7 },
-	{ 0x100076be, 0x01 },
-	{ 0x100076bf, 0x57 },
-	{ 0x100076c0, 0x93 },
-	{ 0x100076c1, 0x07 },
-	{ 0x100076c2, 0xa0 },
-	{ 0x100076c3, 0x05 },
-	{ 0x100076c4, 0x63 },
-	{ 0x100076c5, 0x14 },
-	{ 0x100076c6, 0xf7 },
-	{ 0x100076c7, 0x04 },
-	{ 0x100076c8, 0x37 },
-	{ 0x100076c9, 0x07 },
-	{ 0x100076ca, 0x00 },
-	{ 0x100076cb, 0x11 },
-	{ 0x100076cc, 0x83 },
-	{ 0x100076cd, 0x47 },
-	{ 0x100076ce, 0x07 },
-	{ 0x100076cf, 0x01 },
-	{ 0x100076d0, 0x13 },
-	{ 0x100076d1, 0x06 },
-	{ 0x100076d2, 0x30 },
-	{ 0x100076d3, 0x00 },
-	{ 0x100076d4, 0x93 },
-	{ 0x100076d5, 0xf7 },
-	{ 0x100076d6, 0xf7 },
-	{ 0x100076d7, 0x0f },
-	{ 0x100076d8, 0x63 },
-	{ 0x100076d9, 0x9a },
-	{ 0x100076da, 0xc7 },
-	{ 0x100076db, 0x02 },
-	{ 0x100076dc, 0x03 },
-	{ 0x100076dd, 0x47 },
-	{ 0x100076de, 0x87 },
-	{ 0x100076df, 0x01 },
-	{ 0x100076e0, 0x13 },
-	{ 0x100076e1, 0x77 },
-	{ 0x100076e2, 0xf7 },
-	{ 0x100076e3, 0x0f },
-	{ 0x100076e4, 0x63 },
-	{ 0x100076e5, 0x14 },
-	{ 0x100076e6, 0xf7 },
-	{ 0x100076e7, 0x02 },
-	{ 0x100076e8, 0x37 },
-	{ 0x100076e9, 0xd7 },
-	{ 0x100076ea, 0x00 },
-	{ 0x100076eb, 0x00 },
-	{ 0x100076ec, 0x83 },
-	{ 0x100076ed, 0x47 },
-	{ 0x100076ee, 0x37 },
-	{ 0x100076ef, 0x54 },
-	{ 0x100076f0, 0x93 },
-	{ 0x100076f1, 0xf7 },
-	{ 0x100076f2, 0xf7 },
-	{ 0x100076f3, 0x0f },
-	{ 0x100076f4, 0x93 },
-	{ 0x100076f5, 0xe7 },
-	{ 0x100076f6, 0x07 },
-	{ 0x100076f7, 0x02 },
-	{ 0x100076f8, 0xa3 },
-	{ 0x100076f9, 0x01 },
-	{ 0x100076fa, 0xf7 },
-	{ 0x100076fb, 0x54 },
-	{ 0x100076fc, 0x83 },
-	{ 0x100076fd, 0x47 },
-	{ 0x100076fe, 0x37 },
-	{ 0x100076ff, 0x54 },
-	{ 0x10007700, 0x93 },
-	{ 0x10007701, 0xf7 },
-	{ 0x10007702, 0xf7 },
-	{ 0x10007703, 0x0d },
-	{ 0x10007704, 0xa3 },
-	{ 0x10007705, 0x01 },
-	{ 0x10007706, 0xf7 },
-	{ 0x10007707, 0x54 },
-	{ 0x10007708, 0x23 },
-	{ 0x10007709, 0xa8 },
-	{ 0x1000770a, 0x01 },
-	{ 0x1000770b, 0x56 },
-	{ 0x1000770c, 0xb7 },
-	{ 0x1000770d, 0xd7 },
-	{ 0x1000770e, 0x00 },
-	{ 0x1000770f, 0x10 },
-	{ 0x10007710, 0x03 },
-	{ 0x10007711, 0xc7 },
-	{ 0x10007712, 0x07 },
-	{ 0x10007713, 0xd9 },
-	{ 0x10007714, 0x93 },
-	{ 0x10007715, 0x06 },
-	{ 0x10007716, 0x10 },
-	{ 0x10007717, 0x00 },
-	{ 0x10007718, 0x13 },
-	{ 0x10007719, 0x77 },
-	{ 0x1000771a, 0xf7 },
-	{ 0x1000771b, 0x0f },
-	{ 0x1000771c, 0x63 },
-	{ 0x1000771d, 0x1a },
-	{ 0x1000771e, 0xd7 },
-	{ 0x1000771f, 0x04 },
-	{ 0x10007720, 0x03 },
-	{ 0x10007721, 0xc7 },
-	{ 0x10007722, 0x27 },
-	{ 0x10007723, 0xd9 },
-	{ 0x10007724, 0x13 },
-	{ 0x10007725, 0x07 },
-	{ 0x10007726, 0x17 },
-	{ 0x10007727, 0x00 },
-	{ 0x10007728, 0x13 },
-	{ 0x10007729, 0x77 },
-	{ 0x1000772a, 0xf7 },
-	{ 0x1000772b, 0x0f },
-	{ 0x1000772c, 0x23 },
-	{ 0x1000772d, 0x89 },
-	{ 0x1000772e, 0xe7 },
-	{ 0x1000772f, 0xd8 },
-	{ 0x10007730, 0x83 },
-	{ 0x10007731, 0xc6 },
-	{ 0x10007732, 0x27 },
-	{ 0x10007733, 0xd9 },
-	{ 0x10007734, 0x03 },
-	{ 0x10007735, 0xc7 },
-	{ 0x10007736, 0x17 },
-	{ 0x10007737, 0xd9 },
-	{ 0x10007738, 0x93 },
-	{ 0x10007739, 0xf6 },
-	{ 0x1000773a, 0xf6 },
-	{ 0x1000773b, 0x0f },
-	{ 0x1000773c, 0x13 },
-	{ 0x1000773d, 0x77 },
-	{ 0x1000773e, 0xf7 },
-	{ 0x1000773f, 0x0f },
-	{ 0x10007740, 0x63 },
-	{ 0x10007741, 0xe8 },
-	{ 0x10007742, 0xe6 },
-	{ 0x10007743, 0x02 },
-	{ 0x10007744, 0xb7 },
-	{ 0x10007745, 0xd6 },
-	{ 0x10007746, 0x00 },
-	{ 0x10007747, 0x00 },
-	{ 0x10007748, 0x03 },
-	{ 0x10007749, 0xc7 },
-	{ 0x1000774a, 0xa6 },
-	{ 0x1000774b, 0xe1 },
-	{ 0x1000774c, 0x13 },
-	{ 0x1000774d, 0x67 },
-	{ 0x1000774e, 0x07 },
-	{ 0x1000774f, 0xf8 },
-	{ 0x10007750, 0x13 },
-	{ 0x10007751, 0x77 },
-	{ 0x10007752, 0xf7 },
-	{ 0x10007753, 0x0f },
-	{ 0x10007754, 0x23 },
-	{ 0x10007755, 0x8d },
-	{ 0x10007756, 0xe6 },
-	{ 0x10007757, 0xe0 },
-	{ 0x10007758, 0x03 },
-	{ 0x10007759, 0xc7 },
-	{ 0x1000775a, 0x37 },
-	{ 0x1000775b, 0xd9 },
-	{ 0x1000775c, 0x13 },
-	{ 0x1000775d, 0x07 },
-	{ 0x1000775e, 0x17 },
-	{ 0x1000775f, 0x00 },
-	{ 0x10007760, 0x13 },
-	{ 0x10007761, 0x77 },
-	{ 0x10007762, 0xf7 },
-	{ 0x10007763, 0x0f },
-	{ 0x10007764, 0xa3 },
-	{ 0x10007765, 0x89 },
-	{ 0x10007766, 0xe7 },
-	{ 0x10007767, 0xd8 },
-	{ 0x10007768, 0x13 },
-	{ 0x10007769, 0x07 },
-	{ 0x1000776a, 0x20 },
-	{ 0x1000776b, 0x00 },
-	{ 0x1000776c, 0x23 },
-	{ 0x1000776d, 0x88 },
-	{ 0x1000776e, 0xe7 },
-	{ 0x1000776f, 0xd8 },
-	{ 0x10007770, 0x83 },
-	{ 0x10007771, 0x20 },
-	{ 0x10007772, 0xc1 },
-	{ 0x10007773, 0x00 },
-	{ 0x10007774, 0x03 },
-	{ 0x10007775, 0x24 },
-	{ 0x10007776, 0x81 },
-	{ 0x10007777, 0x00 },
-	{ 0x10007778, 0x13 },
-	{ 0x10007779, 0x01 },
-	{ 0x1000777a, 0x01 },
-	{ 0x1000777b, 0x01 },
-	{ 0x1000777c, 0x67 },
-	{ 0x1000777d, 0x80 },
-	{ 0x1000777e, 0x00 },
-	{ 0x1000777f, 0x00 },
-	{ 0x10007780, 0x03 },
-	{ 0x10007781, 0xc7 },
-	{ 0x10007782, 0xa1 },
-	{ 0x10007783, 0x40 },
-	{ 0x10007784, 0x93 },
-	{ 0x10007785, 0x06 },
-	{ 0x10007786, 0x10 },
-	{ 0x10007787, 0x00 },
-	{ 0x10007788, 0x63 },
-	{ 0x10007789, 0x16 },
-	{ 0x1000778a, 0xd7 },
-	{ 0x1000778b, 0x00 },
-	{ 0x1000778c, 0xb7 },
-	{ 0x1000778d, 0xd6 },
-	{ 0x1000778e, 0x00 },
-	{ 0x1000778f, 0x10 },
-	{ 0x10007790, 0xa3 },
-	{ 0x10007791, 0x8a },
-	{ 0x10007792, 0xe6 },
-	{ 0x10007793, 0xd8 },
-	{ 0x10007794, 0x83 },
-	{ 0x10007795, 0xc7 },
-	{ 0x10007796, 0xa1 },
-	{ 0x10007797, 0x40 },
-	{ 0x10007798, 0x63 },
-	{ 0x10007799, 0x9c },
-	{ 0x1000779a, 0x07 },
-	{ 0x1000779b, 0x06 },
-	{ 0x1000779c, 0x13 },
-	{ 0x1000779d, 0x01 },
-	{ 0x1000779e, 0x01 },
-	{ 0x1000779f, 0xff },
-	{ 0x100077a0, 0x23 },
-	{ 0x100077a1, 0x22 },
-	{ 0x100077a2, 0x91 },
-	{ 0x100077a3, 0x00 },
-	{ 0x100077a4, 0x23 },
-	{ 0x100077a5, 0x26 },
-	{ 0x100077a6, 0x11 },
-	{ 0x100077a7, 0x00 },
-	{ 0x100077a8, 0x23 },
-	{ 0x100077a9, 0x24 },
-	{ 0x100077aa, 0x81 },
-	{ 0x100077ab, 0x00 },
-	{ 0x100077ac, 0xb7 },
-	{ 0x100077ad, 0xc4 },
-	{ 0x100077ae, 0x00 },
-	{ 0x100077af, 0x00 },
-	{ 0x100077b0, 0x83 },
-	{ 0x100077b1, 0xc7 },
-	{ 0x100077b2, 0x04 },
-	{ 0x100077b3, 0x56 },
-	{ 0x100077b4, 0x13 },
-	{ 0x100077b5, 0x07 },
-	{ 0x100077b6, 0x80 },
-	{ 0x100077b7, 0x01 },
-	{ 0x100077b8, 0x93 },
-	{ 0x100077b9, 0xf7 },
-	{ 0x100077ba, 0xf7 },
-	{ 0x100077bb, 0x0f },
-	{ 0x100077bc, 0x63 },
-	{ 0x100077bd, 0x70 },
-	{ 0x100077be, 0xf7 },
-	{ 0x100077bf, 0x04 },
-	{ 0x100077c0, 0x37 },
-	{ 0x100077c1, 0xd4 },
-	{ 0x100077c2, 0x00 },
-	{ 0x100077c3, 0x10 },
-	{ 0x100077c4, 0x83 },
-	{ 0x100077c5, 0x47 },
-	{ 0x100077c6, 0x54 },
-	{ 0x100077c7, 0xd9 },
-	{ 0x100077c8, 0x93 },
-	{ 0x100077c9, 0xf7 },
-	{ 0x100077ca, 0xf7 },
-	{ 0x100077cb, 0x0f },
-	{ 0x100077cc, 0x63 },
-	{ 0x100077cd, 0x88 },
-	{ 0x100077ce, 0x07 },
-	{ 0x100077cf, 0x02 },
-	{ 0x100077d0, 0x93 },
-	{ 0x100077d1, 0x07 },
-	{ 0x100077d2, 0x10 },
-	{ 0x100077d3, 0x00 },
-	{ 0x100077d4, 0x23 },
-	{ 0x100077d5, 0x82 },
-	{ 0x100077d6, 0xf4 },
-	{ 0x100077d7, 0x58 },
-	{ 0x100077d8, 0x03 },
-	{ 0x100077d9, 0x45 },
-	{ 0x100077da, 0x64 },
-	{ 0x100077db, 0xd9 },
-	{ 0x100077dc, 0xb7 },
-	{ 0x100077dd, 0x15 },
-	{ 0x100077de, 0x00 },
-	{ 0x100077df, 0x00 },
-	{ 0x100077e0, 0x93 },
-	{ 0x100077e1, 0x85 },
-	{ 0x100077e2, 0x85 },
-	{ 0x100077e3, 0x38 },
-	{ 0x100077e4, 0x13 },
-	{ 0x100077e5, 0x75 },
-	{ 0x100077e6, 0xf5 },
-	{ 0x100077e7, 0x0f },
-	{ 0x100077e8, 0xef },
-	{ 0x100077e9, 0xe0 },
-	{ 0x100077ea, 0x9f },
-	{ 0x100077eb, 0xd0 },
-	{ 0x100077ec, 0x93 },
-	{ 0x100077ed, 0x55 },
-	{ 0x100077ee, 0xf5 },
-	{ 0x100077ef, 0x41 },
-	{ 0x100077f0, 0xef },
-	{ 0x100077f1, 0xe0 },
-	{ 0x100077f2, 0x8f },
-	{ 0x100077f3, 0xa3 },
-	{ 0x100077f4, 0x23 },
-	{ 0x100077f5, 0x82 },
-	{ 0x100077f6, 0x04 },
-	{ 0x100077f7, 0x58 },
-	{ 0x100077f8, 0xa3 },
-	{ 0x100077f9, 0x0a },
-	{ 0x100077fa, 0x04 },
-	{ 0x100077fb, 0xd8 },
-	{ 0x100077fc, 0x83 },
-	{ 0x100077fd, 0x20 },
-	{ 0x100077fe, 0xc1 },
-	{ 0x100077ff, 0x00 },
-	{ 0x10007800, 0x03 },
-	{ 0x10007801, 0x24 },
-	{ 0x10007802, 0x81 },
-	{ 0x10007803, 0x00 },
-	{ 0x10007804, 0x83 },
-	{ 0x10007805, 0x24 },
-	{ 0x10007806, 0x41 },
-	{ 0x10007807, 0x00 },
-	{ 0x10007808, 0x13 },
-	{ 0x10007809, 0x01 },
-	{ 0x1000780a, 0x01 },
-	{ 0x1000780b, 0x01 },
-	{ 0x1000780c, 0x67 },
-	{ 0x1000780d, 0x80 },
-	{ 0x1000780e, 0x00 },
-	{ 0x1000780f, 0x00 },
-	{ 0x10007810, 0x67 },
-	{ 0x10007811, 0x80 },
-	{ 0x10007812, 0x00 },
-	{ 0x10007813, 0x00 },
-	{ 0x10007814, 0x13 },
-	{ 0x10007815, 0x01 },
-	{ 0x10007816, 0x01 },
-	{ 0x10007817, 0xff },
-	{ 0x10007818, 0x23 },
-	{ 0x10007819, 0x26 },
-	{ 0x1000781a, 0x11 },
-	{ 0x1000781b, 0x00 },
-	{ 0x1000781c, 0xef },
-	{ 0x1000781d, 0xd0 },
-	{ 0x1000781e, 0x8f },
-	{ 0x1000781f, 0x86 },
-	{ 0x10007820, 0x83 },
-	{ 0x10007821, 0xc7 },
-	{ 0x10007822, 0x11 },
-	{ 0x10007823, 0x42 },
-	{ 0x10007824, 0x63 },
-	{ 0x10007825, 0x86 },
-	{ 0x10007826, 0x07 },
-	{ 0x10007827, 0x00 },
-	{ 0x10007828, 0x03 },
-	{ 0x10007829, 0xc7 },
-	{ 0x1000782a, 0x01 },
-	{ 0x1000782b, 0x42 },
-	{ 0x1000782c, 0x63 },
-	{ 0x1000782d, 0x10 },
-	{ 0x1000782e, 0x07 },
-	{ 0x1000782f, 0x02 },
-	{ 0x10007830, 0x83 },
-	{ 0x10007831, 0xc6 },
-	{ 0x10007832, 0x21 },
-	{ 0x10007833, 0x41 },
-	{ 0x10007834, 0x13 },
-	{ 0x10007835, 0x07 },
-	{ 0x10007836, 0xf0 },
-	{ 0x10007837, 0x01 },
-	{ 0x10007838, 0x13 },
-	{ 0x10007839, 0x05 },
-	{ 0x1000783a, 0xf0 },
-	{ 0x1000783b, 0x01 },
-	{ 0x1000783c, 0x63 },
-	{ 0x1000783d, 0x98 },
-	{ 0x1000783e, 0xe6 },
-	{ 0x1000783f, 0x02 },
-	{ 0x10007840, 0x63 },
-	{ 0x10007841, 0x8a },
-	{ 0x10007842, 0x07 },
-	{ 0x10007843, 0x02 },
-	{ 0x10007844, 0x83 },
-	{ 0x10007845, 0xc7 },
-	{ 0x10007846, 0x01 },
-	{ 0x10007847, 0x42 },
-	{ 0x10007848, 0x63 },
-	{ 0x10007849, 0x86 },
-	{ 0x1000784a, 0x07 },
-	{ 0x1000784b, 0x02 },
-	{ 0x1000784c, 0x83 },
-	{ 0x1000784d, 0xc7 },
-	{ 0x1000784e, 0x31 },
-	{ 0x1000784f, 0x42 },
-	{ 0x10007850, 0x63 },
-	{ 0x10007851, 0x86 },
-	{ 0x10007852, 0x07 },
-	{ 0x10007853, 0x00 },
-	{ 0x10007854, 0x83 },
-	{ 0x10007855, 0xc7 },
-	{ 0x10007856, 0x21 },
-	{ 0x10007857, 0x42 },
-	{ 0x10007858, 0x63 },
-	{ 0x10007859, 0x9e },
-	{ 0x1000785a, 0x07 },
-	{ 0x1000785b, 0x00 },
-	{ 0x1000785c, 0x03 },
-	{ 0x1000785d, 0xc7 },
-	{ 0x1000785e, 0x21 },
-	{ 0x1000785f, 0x41 },
-	{ 0x10007860, 0x93 },
-	{ 0x10007861, 0x07 },
-	{ 0x10007862, 0xb0 },
-	{ 0x10007863, 0x01 },
-	{ 0x10007864, 0x63 },
-	{ 0x10007865, 0x08 },
-	{ 0x10007866, 0xf7 },
-	{ 0x10007867, 0x00 },
-	{ 0x10007868, 0x13 },
-	{ 0x10007869, 0x05 },
-	{ 0x1000786a, 0xb0 },
-	{ 0x1000786b, 0x01 },
-	{ 0x1000786c, 0xef },
-	{ 0x1000786d, 0xd0 },
-	{ 0x1000786e, 0x0f },
-	{ 0x1000786f, 0xcf },
-	{ 0x10007870, 0xef },
-	{ 0x10007871, 0xd0 },
-	{ 0x10007872, 0x8f },
-	{ 0x10007873, 0xa4 },
-	{ 0x10007874, 0x93 },
-	{ 0x10007875, 0x06 },
-	{ 0x10007876, 0x10 },
-	{ 0x10007877, 0x00 },
-	{ 0x10007878, 0xa3 },
-	{ 0x10007879, 0x89 },
-	{ 0x1000787a, 0xd1 },
-	{ 0x1000787b, 0x40 },
-	{ 0x1000787c, 0x37 },
-	{ 0x1000787d, 0xd7 },
-	{ 0x1000787e, 0x00 },
-	{ 0x1000787f, 0x10 },
-	{ 0x10007880, 0x83 },
-	{ 0x10007881, 0x47 },
-	{ 0x10007882, 0x07 },
-	{ 0x10007883, 0xd9 },
-	{ 0x10007884, 0x93 },
-	{ 0x10007885, 0xf7 },
-	{ 0x10007886, 0xf7 },
-	{ 0x10007887, 0x0f },
-	{ 0x10007888, 0x63 },
-	{ 0x10007889, 0x90 },
-	{ 0x1000788a, 0x07 },
-	{ 0x1000788b, 0x02 },
-	{ 0x1000788c, 0x37 },
-	{ 0x1000788d, 0xc6 },
-	{ 0x1000788e, 0x00 },
-	{ 0x1000788f, 0x00 },
-	{ 0x10007890, 0x83 },
-	{ 0x10007891, 0x47 },
-	{ 0x10007892, 0x26 },
-	{ 0x10007893, 0x04 },
-	{ 0x10007894, 0x93 },
-	{ 0x10007895, 0xe7 },
-	{ 0x10007896, 0x07 },
-	{ 0x10007897, 0xf8 },
-	{ 0x10007898, 0x93 },
-	{ 0x10007899, 0xf7 },
-	{ 0x1000789a, 0xf7 },
-	{ 0x1000789b, 0x0f },
-	{ 0x1000789c, 0x23 },
-	{ 0x1000789d, 0x01 },
-	{ 0x1000789e, 0xf6 },
-	{ 0x1000789f, 0x04 },
-	{ 0x100078a0, 0x23 },
-	{ 0x100078a1, 0x08 },
-	{ 0x100078a2, 0xd7 },
-	{ 0x100078a3, 0xd8 },
-	{ 0x100078a4, 0x23 },
-	{ 0x100078a5, 0x09 },
-	{ 0x100078a6, 0x07 },
-	{ 0x100078a7, 0xd8 },
-	{ 0x100078a8, 0x83 },
-	{ 0x100078a9, 0x20 },
-	{ 0x100078aa, 0xc1 },
-	{ 0x100078ab, 0x00 },
-	{ 0x100078ac, 0x13 },
-	{ 0x100078ad, 0x01 },
-	{ 0x100078ae, 0x01 },
-	{ 0x100078af, 0x01 },
-	{ 0x100078b0, 0x67 },
-	{ 0x100078b1, 0x80 },
-	{ 0x100078b2, 0x00 },
-	{ 0x100078b3, 0x00 },
-	{ 0x3fc2bfc7, 0x00 },
-	{ 0x3fc2bfc6, 0x00 },
-	{ 0x3fc2bfc5, 0x00 },
-	{ 0x3fc2bfc4, 0x01 },
-	{ 0x0000d486, 0x43 },
-	{ 0x1000db00, 0x02 },
-	{ 0x1000db01, 0x00 },
-	{ 0x1000db02, 0x11 },
-	{ 0x1000db03, 0x00 },
-	{ 0x1000db04, 0x00 },
-	{ 0x1000db05, 0x82 },
-	{ 0x1000db06, 0x04 },
-	{ 0x1000db07, 0xf1 },
-	{ 0x1000db08, 0x00 },
-	{ 0x1000db09, 0x00 },
-	{ 0x1000db0a, 0x40 },
-	{ 0x0000d540, 0x01 },
-};
-
 static const struct reg_default rt1320_reg_defaults[] = {
 	{ SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT_FU21, RT1320_SDCA_CTL_FU_MUTE, CH_01), 0x01 },
 	{ SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT_FU21, RT1320_SDCA_CTL_FU_MUTE, CH_02), 0x01 },
@@ -3733,17 +525,88 @@ static int rt1320_pde_transition_delay(struct rt1320_sdw_priv *rt1320, unsigned
 	return 0;
 }
 
+/*
+ * The 'patch code' is written to the patch code area.
+ * The patch code area is used for SDCA register expansion flexibility.
+ */
+static void rt1320_load_mcu_patch(struct rt1320_sdw_priv *rt1320)
+{
+	struct sdw_slave *slave = rt1320->sdw_slave;
+	const struct firmware *patch;
+	const char *filename;
+	unsigned int addr, val;
+	const unsigned char *ptr;
+	int ret, i;
+
+	if (rt1320->version_id <= RT1320_VB)
+		filename = RT1320_VAB_MCU_PATCH;
+	else
+		filename = RT1320_VC_MCU_PATCH;
+
+	/* load the patch code here */
+	ret = request_firmware(&patch, filename, &slave->dev);
+	if (ret) {
+		dev_err(&slave->dev, "%s: Failed to load %s firmware", __func__, filename);
+		regmap_write(rt1320->regmap, 0xc598, 0x00);
+		regmap_write(rt1320->regmap, 0x10007000, 0x67);
+		regmap_write(rt1320->regmap, 0x10007001, 0x80);
+		regmap_write(rt1320->regmap, 0x10007002, 0x00);
+		regmap_write(rt1320->regmap, 0x10007003, 0x00);
+	} else {
+		ptr = (const unsigned char *)patch->data;
+		if ((patch->size % 8) == 0) {
+			for (i = 0; i < patch->size; i += 8) {
+				addr = (ptr[i] & 0xff) | (ptr[i + 1] & 0xff) << 8 |
+					(ptr[i + 2] & 0xff) << 16 | (ptr[i + 3] & 0xff) << 24;
+				val = (ptr[i + 4] & 0xff) | (ptr[i + 5] & 0xff) << 8 |
+					(ptr[i + 6] & 0xff) << 16 | (ptr[i + 7] & 0xff) << 24;
+
+				if (addr > 0x10007ffff || addr < 0x10007000) {
+					dev_err(&slave->dev, "%s: the address 0x%x is wrong", __func__, addr);
+					goto _exit_;
+				}
+				if (val > 0xff) {
+					dev_err(&slave->dev, "%s: the value 0x%x is wrong", __func__, val);
+					goto _exit_;
+				}
+				regmap_write(rt1320->regmap, addr, val);
+			}
+		}
+_exit_:
+		release_firmware(patch);
+	}
+}
+
+static void rt1320_vab_preset(struct rt1320_sdw_priv *rt1320)
+{
+	unsigned int i, reg, val, delay;
+
+	for (i = 0; i < ARRAY_SIZE(rt1320_blind_write); i++) {
+		reg = rt1320_blind_write[i].reg;
+		val = rt1320_blind_write[i].def;
+		delay = rt1320_blind_write[i].delay_us;
+
+		if (reg == 0x3fc2bfc7)
+			rt1320_load_mcu_patch(rt1320);
+
+		regmap_write(rt1320->regmap, reg, val);
+		if (delay)
+			usleep_range(delay, delay + 1000);
+	}
+}
+
 static void rt1320_vc_preset(struct rt1320_sdw_priv *rt1320)
 {
 	struct sdw_slave *slave = rt1320->sdw_slave;
 	unsigned int i, reg, val, delay, retry, tmp;
 
-	regmap_multi_reg_write(rt1320->regmap, rt1320_vc_blind_write, ARRAY_SIZE(rt1320_vc_blind_write));
+	for (i = 0; i < ARRAY_SIZE(rt1320_vc_blind_write); i++) {
+		reg = rt1320_vc_blind_write[i].reg;
+		val = rt1320_vc_blind_write[i].def;
+		delay = rt1320_vc_blind_write[i].delay_us;
 
-	for (i = 0; i < ARRAY_SIZE(rt1320_vc_patch_code_write); i++) {
-		reg = rt1320_vc_patch_code_write[i].reg;
-		val = rt1320_vc_patch_code_write[i].def;
-		delay = rt1320_vc_patch_code_write[i].delay_us;
+		if (reg == 0x3fc2bf83)
+			rt1320_load_mcu_patch(rt1320);
 
 		if ((reg == SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT_PDE23, RT1320_SDCA_CTL_REQ_POWER_STATE, 0)) &&
 			(val == 0x00)) {
@@ -3762,6 +625,9 @@ static void rt1320_vc_preset(struct rt1320_sdw_priv *rt1320)
 		regmap_write(rt1320->regmap, reg, val);
 		if (delay)
 			usleep_range(delay, delay + 1000);
+
+		if (reg == SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT_PDE23, RT1320_SDCA_CTL_REQ_POWER_STATE, 0))
+			rt1320_pde_transition_delay(rt1320, val);
 	}
 }
 
@@ -3799,13 +665,10 @@ static int rt1320_io_init(struct device *dev, struct sdw_slave *slave)
 
 	/* initialization write */
 	if ((amp_func_status & FUNCTION_NEEDS_INITIALIZATION)) {
-		if (rt1320->version_id < RT1320_VC) {
-			regmap_multi_reg_write(rt1320->regmap, rt1320_blind_write, ARRAY_SIZE(rt1320_blind_write));
-			regmap_multi_reg_write(rt1320->regmap, rt1320_patch_code_write,
-				ARRAY_SIZE(rt1320_patch_code_write));
-		} else if (rt1320->version_id == RT1320_VC) {
+		if (rt1320->version_id < RT1320_VC)
+			rt1320_vab_preset(rt1320);
+		else
 			rt1320_vc_preset(rt1320);
-		}
 
 		regmap_write(rt1320->regmap,
 			SDW_SDCA_CTL(FUNC_NUM_AMP, RT1320_SDCA_ENT0, RT1320_SDCA_CTL_FUNC_STATUS, 0),
diff --git a/sound/soc/codecs/rt1320-sdw.h b/sound/soc/codecs/rt1320-sdw.h
index 1fbc1fcd71cf..95ef9d37c6db 100644
--- a/sound/soc/codecs/rt1320-sdw.h
+++ b/sound/soc/codecs/rt1320-sdw.h
@@ -82,6 +82,8 @@ enum rt1320_version_id {
 };
 
 #define RT1320_VER_B_ID 0x07392238
+#define RT1320_VAB_MCU_PATCH "realtek/rt1320/rt1320-patch-code-vab.bin"
+#define RT1320_VC_MCU_PATCH "realtek/rt1320/rt1320-patch-code-vc.bin"
 
 struct rt1320_sdw_priv {
 	struct snd_soc_component *component;
-- 
2.52.0

